---
title: "Final_Project"
author: "Pankaj Gaonkar"
date: "2025-04-10"
output:
  html_document:
    toc: true
    toc_float: true
  pdf_document:
    toc: true
  word_document: default
  md_document:
    variant: gfm
---

# **Final Project _ Script**

## ABOUT
This script is designed for the analysis of the Shotgun_POULTRY microbiome data This script contains the statistical analysis, exploratory plots and publication plots. Each step of the analysis is discussed at my best capabilities.For more detailed understanding please refer the Readme file.
[Click Here: Readme File](https://github.com/ppg0001/PLPA_Assignment/blob/main/README.md)

*NOTE: This document is for class purpose and is intended to be updated*

                            
## Loading packages

```{r loading packages, echo=TRUE, results='hide'}
# Loading Packages
# Note: Uncomment and install/load the packages if required

# if(!requireNamespace("BiocManager")){
#   install.packages("BiocManager")
# }
# BiocManager::install("phyloseq")
# BiocManager::install("microViz")
# BiocManager::install("microbiomeMarker")
# 
# install.packages("remotes")
# remotes::install_github("david-barnett/microViz")
# library(microViz)
# 
# #Install pairwiseAdonis package if not already installed
# if (!requireNamespace("devtools", quietly = TRUE)) {
#   install.packages("devtools")
# }
# devtools::install_github("pmartinezarbizu/pairwiseAdonis/pairwiseAdonis")

# Load the package
# library(pairwiseAdonis)


library(ggplot2)
library(svglite)
library(scales)
library(tibble)
library(reshape2)
library(Polychrome)
library(RColorBrewer)
library(readxl)
library(tidyverse)
library(ggpubr)

library(phyloseq)
library(vegan)
library(microViz)
library(microbiomeMarker)
library(dplyr)
library(tidyr)

library(VennDiagram)
library(grid)

```



## Loading required input files

```{r loading/importing files}

# Load the microbiome realtive abundance data, sample metadata, taxanomy metadata.
# Note: All files are present in project folder and can be uploaded relatively.

tbl_otu <- read.delim("Microbiome_RelAb.tsv") %>%
  column_to_rownames(var = "Taxon")

mt_smpl <- read.delim("Sample_metadata.tsv", 
                      #row.names=1, 
                      stringsAsFactors=TRUE) %>%
           #Duplicating sample column for rowname
           mutate(sample = Sample) %>%
           column_to_rownames(var = "sample")

# Reordering factors
# We have two fain groups: Farm types and Sample types. Within main group we have respective subgroups.

mt_smpl$Farm_type <- factor(mt_smpl$Farm_type,
                         levels = c("Pullet", "Breeder", "Broiler", "Processing_plant")
                         )
mt_smpl$Sample_type <- factor(mt_smpl$Sample_type,
                             levels = c("Litter", "Soil")
                             )
#unique command helps to avoid duplicates, arrange levels as per biological relevance, here based on poultry company and poultry production chain

mt_smpl$Sample_group <- factor(mt_smpl$Sample_group, 
                               levels = unique(c("C2F4L","C2F4L","C2F4L","C2F1L","C3F10L","C3F14L","C2F4S","C2F4S","C2F4S","C2F1S","C3F10S","C3F14S","C2F4F","C2F4F","C2F4F","C4F17L","C4F17L","C4F17L","C4F8L","C4F17S","C4F17S","C4F17S","C4F8S","C4F17F","C4F17F","C4F17F","C8F40L","C8F40L","C8F40L","C6F35L","C6F37L","C6F60L","C7F14L","C7F22L","C7F81L","C8F40S","C8F40S","C8F40S","C6F35S","C6F37S","C6F60S","C7F14S","C7F22S","C7F81S","C8F40F","C8F40F","C8F40F","C4F8PP","C4F8PC","C12FCHL","C13FJBL","C12FCHS","C13FJBS","C14FJOL","C14FJOS","C15FNSL","C15FTCL","C16FHWL","C17FHEL","C18FCLL","C18FHHL","C18FTGL","C20FJAL","C19FTTL","C20FPHL","C15FNSS","C15FTCS","C16FHWS","C17FHES","C18FCLS","C18FHHS","C18FTGS","C20FJAS","C19FTTS","C20PHS")))


mt_smpl$Sample <- factor(mt_smpl$Sample, 
                               levels = c("C2F4L1","C2F4L2","C2F4L3","C2F1L","C3F10L","C3F14L","C2F4S1","C2F4S2","C2F4S3","C2F1S","C3F10S","C3F14S","C2F4F1","C2F4F2","C2F4F3","C4F17L1","C4F17L2","C4F17L3","C4F8L","C4F17S1","C4F17S2","C4F17S3","C4F8S","C4F17F1","C4F17F2","C4F17F3","C8F40L1","C8F40L2","C8F40L3","C6F35L","C6F37L","C6F60L","C7F14L","C7F22L","C7F81L","C8F40S1","C8F40S2","C8F40S3","C6F35S","C6F37S","C6F60S","C7F14S","C7F22S","C7F81S","C8F40F1","C8F40F2","C8F40F3","C4F8PP","C4F8PC","C12FCHL","C13FJBL","C12FCHS","C13FJBS","C14FJOL","C14FJOS","C15FNSL","C15FTCL","C16FHWL","C17FHEL","C18FCLL","C18FHHL","C18FTGL","C20FJAL","C19FTTL","C20FPHL","C15FNSS","C15FTCS","C16FHWS","C17FHES","C18FCLS","C18FHHS","C18FTGS","C20FJAS","C19FTTS","C20PHS"))

# Loading taxonomy metadata file

mt_tax <- read.delim("Taxonomy_metadata.tsv", 
                     #row.names=1, 
                     stringsAsFactors=TRUE) %>%
          column_to_rownames(var = "Taxon")

```


## Setting colors

It is important to follow a consistent coloring scheme throughout the plots and paper. The color palette for the taxa will be created using the package "Polychrome".

```{r color vectors}
# Creating color vectors

# Sample colors
col_sample <- c("#D02C2C", "#5BBCD6", "#F2AD00", "#F98400","#00A08A")

# Source colors
col_farm <- c("#E0BD48", "#319dc8", "#735794", "darkgrey" )

col_phy <- c ("#0DFFCA","#DDE996","#FB6CE0","#71BDA3","#FFAE8D","#A90040","#C2C8FE","#72722A","#D1EE0D","#FD89BB","#1CFE1C","#FD00FB","#165A6A","#79351C","#9E00FF","#E8DAD4","#26FAFD","#FF00B1","#FF7E1C","#51B44F","#9B4FA7","#F4C538","#2287E2","#ED3B2E","#1C5CFD","#B684FF","#835D75","#535454")

col_tax <- c("#b2df8a","#f781bf","#a65628","#ffff33","#984ea3","#4daf4a","#ff7f00","#377eb8","#e41a1c","#999999")

# We can use the following as our coloring scheme
swatch(col_sample)
swatch(col_farm)
swatch(col_tax)
swatch(col_phy)
```

## Print dataset structure
```{r print structure}

as_tibble(mt_smpl)

#Note the NA values belong to the Fecal samples and Processing plant samples, which are not included in analysis and will not be included in further analysis for this final project.
 
```


# Metaphlan and phyloseq

- Metaphlan outputs a table of relative abundances. However, phyloseq *only accepts absolute abundances for alpha-diversity analysis*.
  
- In order to use phyloseq the relative abundance data will be coerced into natural counts. This will done by multiplying the relative abundance by the number of mapped reads/sample (included in the table mt_smpl and obtained from metaphlan output), then rounding the table, thus creating a dummy table of natural counts. This table preserves the relative abundances relationships between SGBs and allows phyloseq to run.

## Creating natural count table
      
```{r creating natural count table}

#Creating a vector containing the the total number of reads in the same order as the columns in tbl_otu so they match.

nreads <- mt_smpl$TotalReads
nreads

#Dividing all values in tbl_otu to shrink proportions to 0-1

tbl_otu <- tbl_otu[,1:ncol(tbl_otu)]/100

#Multiplying all values (x) of column ith (i) by the ith element of the nreads vector (i'), such as i(x)*i' = estimated read count (ecount)

tbl_otu_ecount <- as.data.frame(t(t(tbl_otu)*nreads)) %>%
                  dplyr::mutate(across(is.numeric, round))
# The table is being transposed so that every row (i) matches with the vector entry (i'), then performing the multiplication. After this the table is transposed again to original wide shape. Then the multiplied table is fed into the mutate function which goes across and rounds all of the numerical values.

```

## The count data is ready to be imported into phyloseq

**For our analysis, we will focus only on the bacterial community.**

```{r importing files to phyloseq}

#Importing files to phyloseq

OTUcount = otu_table(tbl_otu_ecount, 
                taxa_are_rows = TRUE)
      #Needs to be parsed as matrix.
TAX = tax_table(as.matrix(mt_tax)
                )
      #Needs to be parsed as matrix.
sampledata = sample_data(mt_smpl)

#Creating phyloseq objects
phycount_e = phyloseq(OTUcount, TAX, sampledata)

#Excluding Unclassified, Archaea and Dropping other groups that needs to be excluded
phycount_e <- phycount_e %>%
  subset_taxa(!Phylum == "UNCLASSIFIED") #%>%

```


## phyloseq summary

```{r phyloseq summary}
phycount_e
print("")
print("Unique Phyla in phycount_e Object")
get_taxa_unique(phycount_e, "Phylum")
print("Sample Sources in phycount_e object")
levels(sample_data(phycount_e)$Sample_type)

```

# Alpha Diversity

## Calculating alpha-diversity indexes

```{r rich & diversity}

phycount_div <- estimate_richness(phycount_e, 
                                     split = TRUE, 
                                     measures = c("Observed", "Simpson", "Shannon") 
                                     )

#Adding sample data to table

phycount_div <- left_join(rownames_to_column(phycount_div, "Sample"),
                             mt_smpl,
                             by = "Sample")

#Calculating average and standard deviation

phycount_div <- phycount_div %>%
  group_by(Sample_group) %>%
  mutate(m_Observed = mean(Observed),
         sd_Observed = sd(Observed),
         m_Shannon = mean(Shannon),
         sd_Shannon = sd(Shannon),
         m_Simpson = mean(Simpson),
         sd_Simpson = sd(Simpson)
         )

phycount_div

#Write the alpha diversity values for each sample and average the replicates
write.csv(phycount_div, "Phycount_div_each_sample.csv")


## Table for alpha diversity

Alpha_descriptive_stats <- phycount_div %>%
  filter(Sample_type %in% c("Litter", "Soil")) %>%
  group_by(Farm_type, Sample_type) %>%
  summarise(
         m_Observed = mean(Observed),
         sd_Observed = sd(Observed),
         m_Simpson = mean(Simpson),
         sd_Simpson = sd(Simpson),     
         m_Shannon = mean(Shannon),
         sd_Shannon = sd(Shannon),

  ) %>%
  ungroup()

# Print the table
print(Alpha_descriptive_stats)

write.csv(Alpha_descriptive_stats, "Alpha_Divertsity_descriptive_stats.csv")

```


## Alpha diversity: t-test comparision between Litter and Soil within each farm types

```{r alpha diversity t-test}

# Using t.test to compare between the two groups

##Comparing Observed Richness
print("RICHNESS PULLET")
phycount_div %>%
  filter(Sample_type %in% c("Soil", "Litter"),
         Farm_type == "Pullet") %>%
  t.test(Observed ~ Sample_type, 
       data = .
       )
print("RICHNESS BREEDER")
phycount_div %>%
  filter(Sample_type %in% c("Soil", "Litter"),
         Farm_type == "Breeder") %>%
  t.test(Observed ~ Sample_type, 
       data = .
       )
print("RICHNESS BROILER")
phycount_div %>%
  filter(Sample_type %in% c("Soil", "Litter"),
         Farm_type == "Broiler") %>%
  t.test(Observed ~ Sample_type, 
       data = .
       )


##Comparing Shannon index
print("SHANNON PULLET")
phycount_div %>%
  filter(Sample_type %in% c("Soil", "Litter"),
         Farm_type == "Pullet") %>%
  t.test(Shannon ~ Sample_type, 
       data = .
       )
print("SHANNON BREEDER")
phycount_div %>%
  filter(Sample_type %in% c("Soil", "Litter"),
         Farm_type == "Breeder") %>%
  t.test(Shannon ~ Sample_type, 
       data = .
       )
print("SHANNON BROILER")
phycount_div %>%
  filter(Sample_type %in% c("Soil", "Litter"),
         Farm_type == "Broiler") %>%
  t.test(Shannon ~ Sample_type, 
       data = .
       )

##Comparing Simpson index
print("SIMPSON PULLET")
phycount_div %>%
  filter(Sample_type %in% c("Soil", "Litter"),
         Farm_type == "Pullet") %>%
  t.test(Simpson ~ Sample_type, 
       data = .
       )
print("SIMPSON BREEDER")
phycount_div %>%
  filter(Sample_type %in% c("Soil", "Litter"),
         Farm_type == "Breeder") %>%
  t.test(Simpson ~ Sample_type, 
       data = .
       )
print("SIMPSON BROILER")
phycount_div %>%
  filter(Sample_type %in% c("Soil", "Litter"),
         Farm_type == "Broiler") %>%
  t.test(Observed ~ Sample_type, 
       data = .
       )


```

Writing alpha diversity t-test to file
```{R writing alpha div ttest to file}

## Sink command, it print everything betweeN sinks()

set.seed(43)
sink("AlphaDiversity_ttest.txt")
 
   ##Comparing OBserved Richness
   print("RICHNESS PULLET")
   phycount_div %>%
     filter(Sample_type %in% c("Soil", "Litter"),
            Farm_type == "Pullet") %>%
     t.test(Observed ~ Sample_type, 
          data = .
          )
   print("RICHNESS BREEDER")
   phycount_div %>%
     filter(Sample_type %in% c("Soil", "Litter"),
            Farm_type == "Breeder") %>%
     t.test(Observed ~ Sample_type, 
          data = .
          )
   print("RICHNESS BROILER")
   phycount_div %>%
     filter(Sample_type %in% c("Soil", "Litter"),
            Farm_type == "Broiler") %>%
     t.test(Observed ~ Sample_type, 
          data = .
          )
   
   
   ##Comparing Shannon index
   print("SHANNON PULLET")
   phycount_div %>%
     filter(Sample_type %in% c("Soil", "Litter"),
            Farm_type == "Pullet") %>%
     t.test(Shannon ~ Sample_type, 
          data = .
          )
   print("SHANNON BREEDER")
   phycount_div %>%
     filter(Sample_type %in% c("Soil", "Litter"),
            Farm_type == "Breeder") %>%
    t.test(Shannon ~ Sample_type,
         data = .
         )
  print("SHANNON BROILER")
  phycount_div %>%
    filter(Sample_type %in% c("Soil", "Litter"),
           Farm_type == "Broiler") %>%
    t.test(Shannon ~ Sample_type,
         data = .
         )

  ##Comparing Simpson index
  print("SIMPSON PULLET")
  phycount_div %>%
    filter(Sample_type %in% c("Soil", "Litter"),
           Farm_type == "Pullet") %>%
    t.test(Simpson ~ Sample_type,
         data = .
         )
  print("SIMPSON BREEDER")
  phycount_div %>%
    filter(Sample_type %in% c("Soil", "Litter"),
           Farm_type == "Breeder") %>%
    t.test(Simpson ~ Sample_type,
         data = .
         )
  print("SIMPSON BROILER")
  phycount_div %>%
    filter(Sample_type %in% c("Soil", "Litter"),
           Farm_type == "Broiler") %>%
    t.test(Observed ~ Sample_type,
         data = .
         )

sink()


```


Drawing Richness and Diversity plots
```{r boxplot rich $ diversity}
#Creating vector for comparing samples
 complist <- list(c("Soil", "Litter"))

#Creating vector with new names
 Farm_name <- c("Pullet", "Breeder", "Broiler", "Processing\nPlant")
names(Farm_name) <- c("Pullet", "Breeder", "Broiler", "Processing_plant")

#Drawing Observed Richness plot
bp_rc <- 
      phycount_div %>%
  # filter(Sample_type == "Litter" | Sample_type == "Soil") %>%
  filter(Farm_type != "Processing_plant" & Sample_type != "Fecal") %>%
  ggplot(aes(x = Sample_type,
             y = Observed, 
             fill = Sample_type
             )
         ) +
  geom_boxplot() +
  ggtitle("Observed Richness") +
  theme_bw() +
  theme(axis.text = element_text(size = 12, 
                                 color = "black", 
                                 hjust = 0.5
                                   ),
        axis.text.x = element_text(angle = 45,
                                   vjust = 1,
                                   hjust = 1),
        axis.title = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_line(color = "azure2"),
        plot.title = element_text(size = 14, face = "bold"),
        strip.background = element_rect(fill = "transparent",
                                        color = "transparent"),
        strip.text = element_text(size = 12,
                                  face = "bold"
                                  )
        ) +
  stat_compare_means(comparisons = complist, 
                     method = "t.test", 
                     label = "p.signif", 
                     vjust = 1
                     ) +
  scale_fill_manual(values = col_sample) +
  #facet_grid(Farm_type ~ Company,
   facet_grid(Farm_type ~ .,
             labeller = labeller(Farm_type = Farm_name),
             scales = "free",
            #space = "free_x"
             )



#Drawing Shannon diversity index plot
bp_sn <- 
    phycount_div %>%
  # filter(Sample_type == "Litter" | Sample_type == "Soil") %>%
  filter(Farm_type != "Processing_plant" & Sample_type != "Fecal") %>%
  ggplot(aes(x = Sample_type, 
             y = Shannon, 
             fill = Sample_type
             )
         ) +
  geom_boxplot() +
  ggtitle("Shannon Diversity Index") +
  theme_bw() +
  theme(axis.text = element_text(size = 12, 
                                 color = "black", 
                                 hjust = 0.5
                                   ),
        axis.text.x = element_text(angle = 45,
                                   vjust = 1,
                                   hjust = 1),
        axis.title = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_line(color = "azure2"),
        plot.title = element_text(size = 14, face = "bold"),
        strip.background = element_rect(fill = "transparent",
                                        color = "transparent"),
        strip.text = element_text(size = 12,
                                  face = "bold"
                                  )
        ) +
  stat_compare_means(comparisons = complist, 
                     method = "t.test", 
                     label = "p.signif", 
                     vjust = 1
                     ) +
  scale_fill_manual(values = col_sample) +
  #facet_grid(Farm_type ~ Company,
  facet_grid(Farm_type ~ .,
             labeller = labeller(Farm_type = Farm_name),
             scales = "free",
            space = "free_x"
             )


#Drawing Simpson diversity index plot
bp_sp <- 
  phycount_div %>%
  # filter(Sample_type == "Litter" | Sample_type == "Soil") %>%
  filter(Farm_type != "Processing_plant" & Sample_type != "Fecal") %>%
  ggplot(aes(x = Sample_type,
             y = Simpson, 
             fill = Sample_type
                    )
                ) +
  geom_boxplot() +
  ggtitle("Simpson Diversity Index") +
  theme_bw() +
  theme(axis.text = element_text(size = 12, 
                                 color = "black", 
                                 hjust = 0.5
                                   ),
        axis.text.x = element_text(angle = 45,
                                   vjust = 1,
                                   hjust = 1),
        axis.title = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_line(color = "azure2"),
        plot.title = element_text(size = 14, face = "bold"),
        strip.background = element_rect(fill = "transparent",
                                        color = "transparent"),
        strip.text = element_text(size = 12,
                                  face = "bold"
                                  )
        ) +
  stat_compare_means(comparisons = complist, 
                     method = "t.test", 
                     label = "p.signif", 
                     vjust = 1
                     ) +
  scale_fill_manual(values = col_sample) +
  #facet_grid(Farm_type ~ Company,
  facet_grid(Farm_type ~ .,
             labeller = labeller(Farm_type = Farm_name),
             scales = "free",
            space = "free_x"
             )


```  

```{r drawing alpha diversity box plots}
bp_rc
bp_sn
bp_sp
```

```{r saving Alpha Diversity plots}
#Saving plots

#Combined plots
combined_plot_Alpha_Diversity <- ggarrange(bp_rc, bp_sn, bp_sp,
                           ncol = 3,
                           nrow = 1,
                           common.legend = TRUE
)


#Saving plots
ggsave(filename = "Alpha_Diversity_Plots.svg",
       plot = combined_plot_Alpha_Diversity,
       device = "svg",
       units = "mm",
       width = 300,
       height = 250)


```

## Alpha diversity result explanation:
In our study, observed richness and Simpson diversity index were higher in the soil during the pullet and broiler stages, whereas no significant differences were found between litter and soil in the breeder stage. Moreover, the Shannon diversity index was higher in soil samples than in litter across all production stages. These findings indicate that soil contains a more diverse and evenly distributed microbiome, whereas litter is a more selective environment with limited microbes that make up its composition.


# Exploring the microbial community

## Lets check which phylum are overall present in our dataset

```{r metaphlan taxa}
table(tax_table(phycount_e)[,"Phylum"])
```


  
## Microbial community bar plots

First step in visualizing the community structure. In order to create the barplots, the data needs to be transformed from wide to long. This will also allow all the metadata to be incorporated into the table

```{r creating data frame}

#Melting table

tbl_mstr_pc <- melt(rownames_to_column(tbl_otu, var = "Taxon"),
                 id.vars = c("Taxon"),
                 variable.name = "Sample",
                 value.name = "Count") %>%
#Adding Metadata
            left_join(mt_smpl,
                      by = "Sample") %>%
            left_join(rownames_to_column(mt_tax, var = "Taxon"),
                      by = "Taxon")

```

Re-ordering factors so that the main colors of the swatch correspond to the most abundant phyla across:

```{r reordering}
#Summarizing counts by phyla

tbl_mstr_pc %>% 
  group_by(Phylum) %>% 
  summarise(Count = sum(Count)) %>% 
  arrange(Count) %>%
  select(Phylum) %>%
  ungroup()

#Reordering factors in Phylum column according to most abundant.
#NOTE: Phyla will be reordered in ascending order (from least to most). This allows us to drop some levels (ex: Unclassified), without changing the color key for the other levels

tbl_mstr_pc$Phylum <- factor(tbl_mstr_pc$Phylum, 
                          levels = c("Deferribacteres","Rhodothermaeota","Candidatus_Adlerbacteria","Candidatus_Saccharibacteria","Lentisphaerae","Spirochaetes","Chlamydiae","Synergistetes","Verrucomicrobia","Candidatus_Thermoplasmatota","Gemmatimonadetes","Nitrospirae","Candidatus_Melainabacteria","Bacteria_unclassified","Fusobacteria","Acidobacteria","Planctomycetes","Chloroflexi","Deinococcus_Thermus","Cyanobacteria","Bacteroidetes","Proteobacteria","Actinobacteria","Firmicutes","Ascomycota","Thaumarchaeota","Euryarchaeota","UNCLASSIFIED"))

```

Creating summarised table for barplots
  
```{r creating barplot table}

tbl_mstr_bp <- tbl_mstr_pc %>%
  group_by(Sample, Phylum) %>%
  summarise(Count = sum(Count)
            ) %>%
  left_join(mt_smpl, by = "Sample") %>%
  distinct() %>%
  droplevels() %>%
  ungroup

```


```{r creating averaged table}
# Averaging taxons by Sample_group

tbl_mstr_av <- tbl_mstr_pc %>%
  group_by(Taxon, Sample_group) %>%
  summarise(Count = mean(Count)) %>%
  left_join(select(mt_smpl, !c(Sample, TotalReads)), by = "Sample_group") %>%
  left_join(rownames_to_column(mt_tax, var = "Taxon"), by = "Taxon") %>%
  distinct() %>%
  droplevels() %>%
  ungroup()



##Summarising by Phylum count

tbl_mstr_av_bp <- tbl_mstr_av %>%
  group_by(Phylum, Sample_group) %>%
  summarise(Count = sum(Count)) %>%
  left_join(select(mt_smpl, !c(Sample, TotalReads)), by = "Sample_group") %>%
  distinct() %>%
  droplevels() %>%
  ungroup()

#Reordering Phylum factors (left join brought in factors with disordered levels)

tbl_mstr_av$Phylum <- factor(tbl_mstr_av$Phylum, 
                          levels = c("Deferribacteres","Rhodothermaeota","Candidatus_Adlerbacteria","Candidatus_Saccharibacteria","Lentisphaerae","Spirochaetes","Chlamydiae","Synergistetes","Verrucomicrobia","Candidatus_Thermoplasmatota","Gemmatimonadetes","Nitrospirae","Candidatus_Melainabacteria","Bacteria_unclassified","Fusobacteria","Acidobacteria","Planctomycetes","Chloroflexi","Deinococcus_Thermus","Cyanobacteria","Bacteroidetes","Proteobacteria","Actinobacteria","Firmicutes","Ascomycota","Thaumarchaeota","Euryarchaeota","UNCLASSIFIED"))




tbl_mstr_av_bp$Phylum <- factor(tbl_mstr_av_bp$Phylum, 
                           levels = c("Deferribacteres","Rhodothermaeota","Candidatus_Adlerbacteria","Candidatus_Saccharibacteria","Lentisphaerae","Spirochaetes","Chlamydiae","Synergistetes","Verrucomicrobia","Candidatus_Thermoplasmatota","Gemmatimonadetes","Nitrospirae","Candidatus_Melainabacteria","Bacteria_unclassified","Fusobacteria","Acidobacteria","Planctomycetes","Chloroflexi","Deinococcus_Thermus","Cyanobacteria","Bacteroidetes","Proteobacteria","Actinobacteria","Firmicutes","Ascomycota","Thaumarchaeota","Euryarchaeota","UNCLASSIFIED"))

```





```{r drawing AVERAGED bar plots}


####Litter bac

relbp_av_bac_Litter <- 
tbl_mstr_av_bp %>%
  subset(Phylum!="UNCLASSIFIED" & Phylum!="Thaumarchaeota" & Phylum!= "Euryarchaeota" & Phylum!= "Ascomycota" & Sample_type == "Litter"  & Farm_type != "Processing_plant") %>%
  ggplot(aes(x = Sample_group, 
             y = Count, 
             fill = Phylum
             )
         ) + 
  geom_bar(position="fill", stat= "identity") +
  scale_y_continuous(labels = percent) +
  scale_fill_manual(values = col_phy) +
  theme_bw() + 
  labs(title = "Microbiome Composition - Bacterial") +
  ylab("Relative Abundance") +
  theme_bw() +
  theme(axis.text = element_text(size = 12, 
                                 color = "black"),
        axis.text.x = element_text(hjust = 1,
                                 vjust = 0.5,
                                 angle = 90), 
        axis.title = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_line(color = "azure2"),
        plot.title = element_text(hjust = 0.5, size = 14, face = "bold"),
        strip.background = element_rect(fill = "transparent",
                                        color = "transparent"),
        strip.text = element_text(size = 12,
                                  face = "bold"
                                  )
        ) +
  facet_grid(Sample_type ~ Farm_type, scale = "free_x", space = "free_x")



####Soil bac

relbp_av_bac_Soil <- 
tbl_mstr_av_bp %>%
  subset(Phylum!="UNCLASSIFIED" & Phylum!="Thaumarchaeota" & Phylum!= "Euryarchaeota" & Phylum!= "Ascomycota" & Sample_type == "Soil"  & Farm_type != "Processing_plant") %>%
  ggplot(aes(x = Sample_group, 
             y = Count, 
             fill = Phylum
             )
         ) + 
  geom_bar(position="fill", stat= "identity") +
  scale_y_continuous(labels = percent) +
  scale_fill_manual(values = col_phy) +
  theme_bw() + 
  ylab("Relative Abundance") +
  theme_bw() +
  theme(axis.text = element_text(size = 12, 
                                 color = "black"),
        axis.text.x = element_text(hjust = 1,
                                 vjust = 0.5,
                                 angle = 90), 
        axis.title = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_line(color = "azure2"),
        strip.background = element_rect(fill = "transparent",
                                        color = "transparent"),
        strip.text = element_text(size = 12,
                                  face = "bold"
                                  )
        ) +
  facet_grid(Sample_type ~ Farm_type, scale = "free_x", space = "free_x")
```

```{r drawing averaged relab plots}
relbp_av_bac_Litter
relbp_av_bac_Soil
```
Saving plots

```{r saving averaged relative abundance plots_Bacteria}

#Saving plots

#Combined plots
combined_plot_relative_abundance_bacteria <- ggarrange(relbp_av_bac_Litter, relbp_av_bac_Soil,
                           ncol = 1,
                           nrow = 2,
                           common.legend = TRUE,
                           legend = "right")

#Saving plots
ggsave(filename = "Relative_abundance_Plots.svg",
       plot = combined_plot_relative_abundance_bacteria,
       device = "svg",
       units = "mm",
       width = 350,
       height = 220)


```



# **Exploring Beta-Diversity**

  If alpha-diversity is the diversity *within* a community, beta-diversity is the diversity across communities. In beta-diversity we are concerned with comparing and contrasting different profiles.

  Communities can be statistically compared by producing distance matrices. On compositional data we will use robust Aitchison distances, which relies on the centered log ratio transform. Once these distances are calculated, profiles can be statistically compared using PERMANOVA and visualized using ordination plots, such as PCoA or PCA.
  
  In order to perform these analysis the abundances will be again exported into phyloseq but this time as relative abundances. The phyloseq object will be used as input into two different packages:
      --> Vegan: where PERMANOVA and PERMAdisp will be calculates
      --> MicrobiomeMarker: where Ordinations and normalization will be performed
      
```{r creating relative abundance phyloseq object}

OTUcount = otu_table(tbl_otu, 
                taxa_are_rows = TRUE)
      #Needs to be parsed as matrix.
TAX = tax_table(as.matrix(mt_tax)
                )
      #Needs to be parsed as matrix.
sampledata = sample_data(mt_smpl)

#Creating phyloseq objects
phycount = phyloseq(OTUcount, TAX, sampledata)


#Excluding Unclassified, Archaea and Dropping incomplete sample groups: Larvae and Pollen
phycount <- phycount %>%
  subset_taxa(!Phylum == "UNCLASSIFIED" & !Kingdom == "Archaea" & !Kingdom == "Eukaryota") %>%
  subset_samples(!Sample_type == "Fecal" & !Farm_type == "Processing_plant")

  

```

```{r printing phyloseq summary}
phycount

print("")
print("Unique Phyla in phycount Object")
get_taxa_unique(phycount, "Phylum")
print("Sample Sources in phycount object")
levels(sample_data(phycount)$Sample_type)
levels(sample_data(phycount)$Farm_type)

```
##Performing PERMANOVAS

- PERMANOVA, (permutational multivariate ANOVA), is a non-parametric alternative to MANOVA, or multivariate ANOVA test.
- PERMANOVAS are calculated using the adonis functions in the package Vegan. To use Vegan, phyloseq objects need to be turned into Vegan objects.
- Next functions convert physeq format into Vegan.

```{r creating pssd2veg function}

pssd2veg <- function(physeq) {
  sd <- sample_data(physeq)
  return(as(sd,"data.frame"))
}

```

```{r creating psotu2veg }

psotu2veg <- function(physeq) {
  OTU <- otu_table(physeq)
  if (taxa_are_rows(OTU)) {
    OTU <- t(OTU)
  }
  return(as(OTU, "matrix"))
}

```

Importing Phyloseq data into teh package Vegan

```{r data import }

vegan_count <- pssd2veg(phycount)
vegan_otu <- psotu2veg(phycount)

```

Calculating distance matrix

```{r distance matrix}
##Calculating distance matrix using Robust Aitchinson (Euclidean distance of the CLR transform of non-zero values)
veg_count_raitch <- vegdist(vegan_otu, "robust.aitchison")


```



 
## RUNNING PERMANOVAs

```{r running distance PERMANOVA}

#Writing the overall PERMANOVA results in text file
sink("PERMANOVA_Overall_Results.txt")


set.seed(43)

print("Comparing profiles by PERMANOVA",quote = FALSE, justify = "centre")
print("Grouping by Sample_group (Sample_type*Farm_type*Company",quote = FALSE, justify = "centre")

adonis2(veg_count_raitch ~ Sample_type*Farm_type*Company, data = vegan_count, permutations = 999)
print("",quote = FALSE, justify = "centre")
print("Grouping by Sample_type",quote = FALSE, justify = "centre")

adonis2(veg_count_raitch ~ Sample_type, data = vegan_count, permutations = 999)
print("",quote = FALSE, justify = "centre")
print("Grouping by Farm_type",quote = FALSE, justify = "centre")

adonis2(veg_count_raitch ~ Farm_type, data = vegan_count, permutations = 999)
print("",quote = FALSE, justify = "centre")
print("Grouping by Company",quote = FALSE, justify = "centre")

adonis2(veg_count_raitch ~ Company, data = vegan_count, permutations = 999)

sink()

```

```{r specific adonis pariwise}

#Writing the specific pariwise comparission of interest in text file
sink("specific_Pairwise_Comparissions_PERMANOVA.txt")

## Install pairwiseAdonis package if not already installed
#if (!requireNamespace("devtools", quietly = TRUE)) {
# install.packages("devtools")
#}
#devtools::install_github("pmartinezarbizu/pairwiseAdonis/pairwiseAdonis")

# Load the package
library(pairwiseAdonis)

# Perform pairwise PERMANOVA for Sample_type
pairwise.adonis2(veg_count_raitch ~ Sample_type, data = vegan_count, permutations = 999)

# Perform pairwise PERMANOVA for Farm_type
pairwise.adonis2(veg_count_raitch ~ Farm_type, data = vegan_count, permutations = 999)


sink()

```



### ORDINATION WITH PCOA

Drawing PCA plots

```{r drawing PCoA + Aithcison plots}

#Sample type
pcoa_sp_clr_S <-
  phycount %>%
 tax_transform(rank = "Species", 
               trans = "identity") %>%  
 dist_calc(dist = "robust.aitchison") %>%
 ord_calc(method = "PCoA") %>% 
 ord_plot(axes = c(1, 2),
          plot_taxa = 1:3,
          colour = "black", 
          fill = "Sample_type",
          shape = "Farm_type",
          alpha = 0.8,
          size = 5
          ) + 
  stat_ellipse(aes(colour = Sample_type), linewidth = 0.3) +
  scale_shape_girafe_filled() +
  ggtitle("PCoA Plot: Beta Diversity - Sample Types") +
   guides(fill = guide_legend(override.aes=list(shape = 21)),
          color = FALSE) +
  scale_fill_manual(values = col_sample) +
  # scale_color_manual(values = col_sample) +
  # scale_alpha_discrete(range = c(0.35, 1)) +
  theme_linedraw() +
  theme(panel.grid = element_blank(),
        panel.background = element_rect(fill = "#fdfdfd"),
        axis.text = element_text(size = 14,
                                 color = "black"),
        axis.title = element_text(size = 16,
                                  color = "black")) +
  geom_text(x = 2.5, 
            y = -3.4, 
            hjust =0,
            vjust = 1,
            label="Sample Type:\nPERMANOVA=0.001",
            size = 3.5,
            fontface = "plain")


#Farm type
pcoa_sp_clr_farm <-
  phycount %>%
 tax_transform(rank = "Species", 
               trans = "identity") %>%  
 dist_calc(dist = "robust.aitchison") %>%
 ord_calc(method = "PCoA") %>% 
 ord_plot(axes = c(1, 2),
          plot_taxa = 1:3,
          colour = "black", 
          fill = "Farm_type",
          shape = "Sample_type",
          alpha = 0.8,
          size = 5
          ) + 
  stat_ellipse(aes(colour = Farm_type), linewidth = 0.3) +
  scale_shape_girafe_filled() +
  ggtitle("PCoA Plot: Beta Diversity - Farm Types") +
   guides(fill = guide_legend(override.aes=list(shape = 21)),
          color = FALSE) +
  scale_fill_manual(values = col_farm) +
  scale_color_manual(values = col_farm) +
  # scale_alpha_discrete(range = c(0.35, 1)) +
  theme_linedraw() +
  theme(panel.grid = element_blank(),
        panel.background = element_rect(fill = "#fdfdfd"),
        axis.text = element_text(size = 14,
                                 color = "black"),
        axis.title = element_text(size = 16,
                                  color = "black")) +
  geom_text(x = 2.5, 
            y = -3.4, 
            hjust =0,
            vjust = 1,
            label="Farm Type:\nPERMANOVA=0.001",
            size = 3.5,
            fontface = "plain") #+
  #facet_grid(. ~ Company, scale = "free")


```



```{r drawing PCoA plots}
pcoa_sp_clr_S

pcoa_sp_clr_farm

```

```{r saving PCoA}
#Saving plots (CAUTION!Run ONLY WHEN READY! Leave commented to avoid overwriting already generated plots)


#Combined plots
combined_PCoA <- ggarrange(pcoa_sp_clr_S, pcoa_sp_clr_farm,
                           ncol = 1,
                           nrow = 2)



 ggsave(filename = "Combined_PCoA_plots_betadiversity.svg",
        plot = combined_PCoA,
        device = "svg",
        units = "mm",
        width = 250,
        height = 300)



```


# Compiling all generated plots as per the figure number


#################################################################################