---
title: "Final_Project"
author: "Pankaj Gaonkar"
date: "2025-04-10"
output:
    md_document:
    variant: gfm
  pdf_document:
    toc: true
  word_document:
  html_document:
    toc: true
    toc_float: true
---

# **Final Project**

#ABOUT

This script is designed for the analysis of the Shotgun_POULTRY microbiome data

This script contains the statistical analysis, exploratory plots and publication plots. Each step of the analysis is discussed at my best capabilities.
NOTE: This document is for class purpose and is intended to be updated

                            

Preparing work environment

```{r loading packages, echo=TRUE, results='hide'}
# Loading Packages
# Note: Uncomment and install/load the packages if required

# if(!requireNamespace("BiocManager")){
#   install.packages("BiocManager")
# }
# BiocManager::install("phyloseq")
# BiocManager::install("microViz")
# BiocManager::install("microbiomeMarker")
# 
# install.packages("remotes")
# remotes::install_github("david-barnett/microViz")
# library(microViz)
# 
# #Install pairwiseAdonis package if not already installed
# if (!requireNamespace("devtools", quietly = TRUE)) {
#   install.packages("devtools")
# }
# devtools::install_github("pmartinezarbizu/pairwiseAdonis/pairwiseAdonis")

# Load the package
# library(pairwiseAdonis)


library(ggplot2)
library(svglite)
library(scales)
library(tibble)
library(reshape2)
library(Polychrome)
library(RColorBrewer)
library(readxl)
library(tidyverse)
library(ggpubr)


library(phyloseq)
library(vegan)
library(microViz)
library(microbiomeMarker)
library(dplyr)
library(tidyr)

#install.packages("UpSetR")
#library(UpSetR)
#install.packages("VennDiagram")
library(VennDiagram)
library(grid)

```



setting work directory and loading necessary input files


```{r loading/importing files}

# Load the microbiome realtive abundance file and sample metadata file.
# Note: All files are present in project folder and can be unploaded relatively.

tbl_otu <- read.delim("Microbiome_RelAb.tsv") %>%
  column_to_rownames(var = "Taxon")

mt_smpl <- read.delim("Sample_metadata.tsv", 
                      #row.names=1, 
                      stringsAsFactors=TRUE) %>%
           #Duplicating sample column for rowname
           mutate(sample = Sample) %>%
           column_to_rownames(var = "sample")

#Reordering factors
mt_smpl$Farm_type <- factor(mt_smpl$Farm_type,
                         levels = c("Pullet", "Breeder", "Broiler", "Processing_plant")
                         )
mt_smpl$Sample_type <- factor(mt_smpl$Sample_type,
                             levels = c("Litter", "Soil", "Post_pick", "Post_chill", "Fecal")
                             )
#unique command helps to avoid duplicates, arrange levels as per biological relevance, here based on company and production chain
mt_smpl$Sample_group <- factor(mt_smpl$Sample_group, 
                               levels = unique(c("C2F4L","C2F4L","C2F4L","C2F1L","C3F10L","C3F14L","C2F4S","C2F4S","C2F4S","C2F1S","C3F10S","C3F14S","C2F4F","C2F4F","C2F4F","C4F17L","C4F17L","C4F17L","C4F8L","C4F17S","C4F17S","C4F17S","C4F8S","C4F17F","C4F17F","C4F17F","C8F40L","C8F40L","C8F40L","C6F35L","C6F37L","C6F60L","C7F14L","C7F22L","C7F81L","C8F40S","C8F40S","C8F40S","C6F35S","C6F37S","C6F60S","C7F14S","C7F22S","C7F81S","C8F40F","C8F40F","C8F40F","C4F8PP","C4F8PC","C12FCHL","C13FJBL","C12FCHS","C13FJBS","C14FJOL","C14FJOS","C15FNSL","C15FTCL","C16FHWL","C17FHEL","C18FCLL","C18FHHL","C18FTGL","C20FJAL","C19FTTL","C20FPHL","C15FNSS","C15FTCS","C16FHWS","C17FHES","C18FCLS","C18FHHS","C18FTGS","C20FJAS","C19FTTS","C20PHS")))


mt_smpl$Sample <- factor(mt_smpl$Sample, 
                               levels = c("C2F4L1","C2F4L2","C2F4L3","C2F1L","C3F10L","C3F14L","C2F4S1","C2F4S2","C2F4S3","C2F1S","C3F10S","C3F14S","C2F4F1","C2F4F2","C2F4F3","C4F17L1","C4F17L2","C4F17L3","C4F8L","C4F17S1","C4F17S2","C4F17S3","C4F8S","C4F17F1","C4F17F2","C4F17F3","C8F40L1","C8F40L2","C8F40L3","C6F35L","C6F37L","C6F60L","C7F14L","C7F22L","C7F81L","C8F40S1","C8F40S2","C8F40S3","C6F35S","C6F37S","C6F60S","C7F14S","C7F22S","C7F81S","C8F40F1","C8F40F2","C8F40F3","C4F8PP","C4F8PC","C12FCHL","C13FJBL","C12FCHS","C13FJBS","C14FJOL","C14FHOS","C15FNSL","C15FTCL","C16FHWL","C17FHEL","C18FCLL","C18FHHL","C18FTGL","C20FJAL","C19FTTL","C20FPHL","C15FNSS","C15FTCS","C16FHWS","C17FHES","C18FCLS","C18FHHS","C18FTGS","C20FJAS","C19FTTS","C20PHS"))



mt_tax <- read.delim("Taxonomy_metadata.tsv", 
                     #row.names=1, 
                     stringsAsFactors=TRUE) %>%
          column_to_rownames(var = "Taxon")

```

Setting colors

It is important to follow a consistent coloring scheme throughout the plots and paper.

Sample:
  --> Litter: "firebrick" (red)
  --> Soil: "#4daf4a" (green)
  --> Feces: excluded
  
Source:
  --> Pullet: "#FC7A16" (light orange)
  --> Breeder: "#319dc8" (blue)
  --> Broiler: "#735794" (purple)
  --> P.Plant & truck: excluded for main
  
The color pallete for the taxa will be created using the package "Polychrome" 

```{r color vectors}
# Creating color vectors

#Sample colors
col_sample <- c("#D02C2C", "#5BBCD6", "#F2AD00", "#F98400","#00A08A")

#Source colors
col_farm <- c("#E0BD48", "#319dc8", "#735794", "darkgrey" )

##Phyla colors with Polycrome
# col_phy <- createPalette(27, c("#F44336", "#2886E2", "#F1C232","#984ea3","#4daf4a","#ff7f00"))
# names(col_phy) <- NULL
# col_phy <- rev(col_phy)
# ##Adding dark grey for UNcLASSIFIED
# col_phy <- append(col_phy,"#535454")

col_phy <- c ("#0DFFCA","#DDE996","#FB6CE0","#71BDA3","#FFAE8D","#A90040","#C2C8FE","#72722A","#D1EE0D","#FD89BB","#1CFE1C","#FD00FB","#165A6A","#79351C","#9E00FF","#E8DAD4","#26FAFD","#FF00B1","#FF7E1C","#51B44F","#9B4FA7","#F4C538","#2287E2","#ED3B2E","#1C5CFD","#B684FF","#835D75","#535454")

col_tax <- c("#b2df8a","#f781bf","#a65628","#ffff33","#984ea3","#4daf4a","#ff7f00","#377eb8","#e41a1c","#999999")



##Phyla colors with Polycrome
col_genus <- createPalette(100, c("#F44336", "#2886E2", "#F1C232","#984ea3","#4daf4a","#ff7f00"))
names(col_genus) <- NULL






swatch(col_sample)
swatch(col_farm)
swatch(col_tax)
swatch(col_phy)
```

#INTRODUCTION

  The demand for producing animal protein has increased worldwide. In consequence, there is an increased in the need of AMU to prevent and treat animal diseases which have contributed to the increase in  AMR and its spread.Because of concerns with AMR,  the animal production industry has promoted reducing AMU to mitigate AMR. However, because of the extensive historic use of antimicrobial in farm, antimicrobial residues might have accumulated in these environments, contributing to the maintenance of AMR. In this study, we explore the between AMR and residues in poultry farms practicing restricted AMU.


#HYPOTHESIS

 We have litter sample representing inside environment and soil representing outside environment We anticipate increase in AMR and residue in inside environment Whereas decrease in outside environment. It might be because of antimicrobial pressure at these points.
 OR
 Although farms in these study practice restricted AMU, because of historic AMU there can be persistence of residue leading to maintenance of AMR.


#OBJECTIVES

  1.Determine AMR levels in the environment of poultry farms in different stages of production. (using qPCR)
    
  2.Determine antimicrobial residues in different stages of poultry production. (using HPLC)
  
  3.*Reconstruct the microbiome and resistome of poultry farm environment (litter,soil) using shotgun metagenomics*.   (using metagenomics)
  
  4.Determine the association of AMR with antimicrobial residues in poultry production.



#MATERIAL AND METHODS


##EXPERIMENTAL DESIGN

  The samples were collected from two different poultry companies (Company A and Company B). The entire poultry complex was screened for each company representing approximately 10% of the respective farm types (pullet, breeder and broiler).
  Total farm numbers (n)= Pullet, breeder, broiler
**Note**: Only include those farms for which we did sequencing (as we did not include all farms for sequencing)
  

    ##SAMPLING COLLECTION
 Farm types: Pullet, breeder, broiler
 Sample types: Litter, soil, post pick , post chill, fecal
 Transportation truck **(EXCLUDED)**: composite fecal samples from base of truck bed (Truck)
 Processing plant**(EXCLUDED)**: Carcass rinses from post pick (PP) and post chill (PC) stage.
 

    ##DNA ISOLATION
  DNA was isolated from Litter, soil, truck, and carcass rinses (PP and PC) using PowerSoil Pro kit (for litter,soil,truck) and PowerWater kit (for carcass rinses)
  
        ###Sample Naming convention: 
  Sample ID convention to identify sequencing groups
      --> C: collection number
      --> F: followed by farm number OR farm name.
          Example:
          F8: farm number eight
          FHE: farm named HE
      --> Letter at end: sample types
          L: litter, S: soil
      --> Example:
          C2F8L: collection 2, farm number 8, litter sample
          C17FHEL: collection 17, farm name HE, litter sample
      --> Note in metadata in some samples they end with number 1,2,or3. They are the triplicate sample we used initially for shotgun sequencing standardization.
      
  
  The whole sampling scheme can be seen the table bellow:  ##### ???????? #####
  
```{r print structure}

##???### Did you already pull Sample_metadata on line 95?
 ##for personal PC
 # smpl_str <- read.delim("C:/Users/panka/Box/Lab shared/Shotgun_poultry_final/MicrobiomeReconstruction/Sample_metadata.tsv", stringsAsFactors=TRUE)
 ##smpl_str <- read.delim("C:/Users/ppg0001/Box/Lab shared/Shotgun_poultry_final/MicrobiomeReconstruction/Sample_metadata.tsv", stringsAsFactors=TRUE)

as_tibble(mt_smpl)
 
```



    ##DNA SEQUENCING AND PROCESSING

The DNA was pair-end sequenced in an Illumina platform. The generated reads were quality checked with FAstQC v0.12.1. 

Reads were processed using Trimmomatic to the following parameters:
  --> Adapter clipping: universal Illumina adapter
  --> End trimming to quality phred 30
  --> Exclusion of resulting reads shorted than 50bp

After processing the reads were again quality checked using FastQC

            -ADD FASTQC DATA ANALYSIS AND DISCUSSION HERE-
            
#MICROBIOME RECONSTRUCTION WITH METAPHLAN4

software/package: MetaPhLAn 4
publication: https://www.nature.com/articles/s41587-023-01688-w
git: https://github.com/biobakery/MetaPhlAn/wiki/MetaPhlAn-4.1
version: 4.0.2

---From git---
                      
  MetaPhlAn is a computational tool for profiling the composition of microbial communities (Bacteria, Archaea and Eukaryotes) from metagenomic shotgun sequencing data (i.e. not 16S) with species-level. With StrainPhlAn, it is possible to perform accurate strain-level microbial profiling.

  MetaPhlAn 4 relies on ~5.1M unique clade-specific marker genes (the latest marker information file can be found here) identified from ~1M microbial genomes (~236,600 references and 771,500 metagenomic assembled genomes) spanning 26,970 species-level genome bins (SGBs, http://segatalab.cibio.unitn.it/data/Pasolli_et_al.html), 4,992 of them taxonomically unidentified at the species level (the full list of the species included in the latest database can be found here), allowing:
  --> unambiguous taxonomic assignments;
  --> an accurate estimation of organismal relative abundance;
  --> SGB-level resolution for bacteria, archaea and eukaryotes;
  --> strain identification and tracking orders of magnitude speedups compared to existing methods.
  --> metagenomic strain-level population genomics

---x---
                             
  Metaphlan produces a table output with RELATIVE ABUNDANCES only. 
This choice by the developers is understandable since Metagenomic shotgun data is, by definition, compositional data.

##COMPOSITIONALITY OF MICROBIOME DATA AND ANALYSIS

The issue of compositionality with microbiome data is explored well by Gloor (2017) at: https://www.frontiersin.org/journals/microbiology/articles/10.3389/fmicb.2017.02224/full

  In ecological studies observations are virtually always independent, i.e.: the ability to observe (count) Tigers does not affect or is affected by the ability to observe ladybugs. Observations (counts) are independent from each other. This assumption governs most of the statistical tools used for ecological data analysis.
  
  However,in HTS studies (such as shotgun metagenomics), the assumption of independence between observations *is not true*. HTS instruments have a fixed number of reads (observations) that can be output no matter the size of the community. This limited number of possible observations means that each count is dependent to every other count. If there's only 1000 observations available, the ability to count tigers is severely impacted by a sudden influx of hundreds of lady bugs.
  
    THE TOTAL READ COUNT OBSERVED IN A HTS RUN IS A FIXED-SIZE RANDOM SAMPLE OF THE *RELATIVE ABUNDANCES* IN THE ECOSYSTEM/COMMUNITY

  Moreover, the read number is constant. It fails to capture every possible features in large communities, however in small communities the same feature is just observed (and counted) over and over again. This is the underlying factor behind sequence coverage, but it also means that there is *no relationship between read number and community size*. A higher coverage simply informs the precision of the observation but does not add any new biological information.
  
    IN HTS STUDIES THE RELATIONSHIP BETWEEN ABSOLUTE ABUNDANCE AND RELATIVE ABUNDANCE IS NOT PREDICTABLE
  
  This issue is compounded in the context of profiling relying in read mapping, since the profile is being drawn from a subset of the read pool, a read pool that is in itself a random sample of relative abundances with no predictable link to community size. It is a "subset" of a "random subset". This makes any attempts to backtrack original absolute abundances virtually impossible. 
  
  The only thing that can be known is the prevalence of a feature in the read pool and it's composition. Here, prevalence is described as:
  
    Prevalence = n feature observations (mapped reads) / Total observations (total reads)
  
  Since the only trustworthy biological information being recovered is relative abundances, microbiome data is more reliable when it is analysed in this context. Working with relative abundances means that  the sums of all observations in a sample always add up to 1. This constrain means the any change in one observation affects all others, since the sum cannot change. Therefore, *microbiome data is compositional and must be analysed as such*

##ALPHA DIVERSITY ANALYSIS WITH PHYLOSEQ

software/package: phyloseq
publication: https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0061217
git: https://joey711.github.io/phyloseq/
version: 1.44.0


  Estimation of diversity within the community. For this two metrics will be calculated:
    1. RICHNESS: simple measure of the number of different Taxon (or OTU) observed in each sample
    2. SIMPSON INDEX (D): measure of diversity (richness x evenness). Calculates the probability of two randomly picked individual being of the same taxon.
        -->Min = 0 : Infinite diversity (zero probability)
        -->Max = 1: No diversity (100% probability)
        OBS: usually expressed as 1-D so the results are more intuitive
    3. SHANNON INDEX: measure of diversity (richness x evenness). Estimates the level of uncertainty that a randomly picked individual belongs to any particular taxon.
        -->Min = 0: No diversity (no uncertainty)
        -->Max = Hmax: Highest diversity (changes depending on S, caution with comparisons)
        
  Metaphlan works with the concept of species-level genome bins (SGBs), which for alpha-diversity is functionally similar to the concept of OTUs in 16S profiling. Each SGB (SGB######) has a taxonomic metadata associated to it (mt_tax) with its taxonomic classification to the lowest possible level.

  As discussed before, Metaphlan outputs a table of relative abundances. However, phyloseq *only accepts absolute abundances for alpha-diversity analysis*.
  
  In order to use phyloseq the relative abundance data will be coerced into natural counts. This will done by multiplying the relative abundance by the number of mapped reads/sample (included in the table mt_smpl and obtained from metaphlan output), then rounding the table, thus creating a dummy table of natural counts. This table preserves the relative abundances relationships between SGBs and allows phyloseq to run. The impact in the metrics is minimal:
      --> RICHNESS: no impact. It's the same for relative or absolute abundance 
      --> Shannon and Simpson: Both transform absolute abundance into relative abundance to calculate the indexes. Negligible effect from eventual rounding.
      
```{r creating natural count table}

#Creating a vector containing the the total number of reads in the same order as the columns in tbl_otu so they match

nreads <- mt_smpl$TotalReads
nreads

#Divinding all values in tbl_otu to shrink proportions to 0-1

tbl_otu <- tbl_otu[,1:ncol(tbl_otu)]/100

#Multiplying all values (x) of column ith (i) by the ith element of the nreads vector (i'), such as i(x)*i' = estimated read count (ecount)

tbl_otu_ecount <- as.data.frame(t(t(tbl_otu)*nreads)) %>%
                  dplyr::mutate(across(is.numeric, round))
# The table is being transposed so that every row (i) matches with the vector entry (i'), then performing the multiplication. After this the table is transposed again to original wide shape. Then the muliplied table is fed into the mutate function which goes across and rounds all of the numerical values.

```

The count data is ready to be imported into phyloseq.

For our analysis, we will focus only on the bacterial community.

```{r importing files to phyloseq}

OTUcount = otu_table(tbl_otu_ecount, 
                taxa_are_rows = TRUE)
      #Needs to be parsed as matrix.
TAX = tax_table(as.matrix(mt_tax)
                )
      #Needs to be parsed as matrix.
sampledata = sample_data(mt_smpl)

#Creating phyloseq objects
phycount_e = phyloseq(OTUcount, TAX, sampledata)
      #Tree doesn't need to be processed. Direct import

#Excluding Unclassified, Archaea and Dropping other groups that needs to be excluded
phycount_e <- phycount_e %>%
  subset_taxa(!Phylum == "UNCLASSIFIED") #%>%
 #subset_samples(!Sample_type == "Fecal")   #DID NIT EXCLUDE PPlant yet

```

##phycount_e ITS OUR NATURAL COUNT


```{r phyloseq summary}
phycount_e
print("")
print("Unique Phyla in phycount_e Object")
get_taxa_unique(phycount_e, "Phylum")
print("Sample Sources in phycount_e object")
levels(sample_data(phycount_e)$Sample_type)

```

Calculating alpha-diversity indexes

```{r rich & diversity}

phycount_div <- estimate_richness(phycount_e, 
                                     split = TRUE, 
                                     measures = c("Observed", "Simpson", "Shannon") 
                                     )

#Adding sample data to table

phycount_div <- left_join(rownames_to_column(phycount_div, "Sample"),
                             mt_smpl,
                             by = "Sample")

#Calculating average and standard deviation

phycount_div <- phycount_div %>%
  group_by(Sample_group) %>%
  mutate(m_Observed = mean(Observed),
         sd_Observed = sd(Observed),
         m_Shannon = mean(Shannon),
         sd_Shannon = sd(Shannon),
         m_Simpson = mean(Simpson),
         sd_Simpson = sd(Simpson)
         )

phycount_div


## Table for alpha diversity

Alpha_descriptive_stats <- phycount_div %>%
  filter(Sample_type %in% c("Litter", "Soil")) %>%
  group_by(Farm_type, Sample_type) %>%
  summarise(
         m_Observed = mean(Observed),
         sd_Observed = sd(Observed),
         m_Simpson = mean(Simpson),
         sd_Simpson = sd(Simpson),     
         m_Shannon = mean(Shannon),
         sd_Shannon = sd(Shannon),

  ) %>%
  ungroup()

# Print the table
print(Alpha_descriptive_stats)





```


```{r alpha diversity t-test}

##Comparing OBserved Richness
print("RICHNESS PULLET")
phycount_div %>%
  filter(Sample_type %in% c("Soil", "Litter"),
         Farm_type == "Pullet") %>%
  t.test(Observed ~ Sample_type, 
       data = .
       )
print("RICHNESS BREEDER")
phycount_div %>%
  filter(Sample_type %in% c("Soil", "Litter"),
         Farm_type == "Breeder") %>%
  t.test(Observed ~ Sample_type, 
       data = .
       )
print("RICHNESS BROILER")
phycount_div %>%
  filter(Sample_type %in% c("Soil", "Litter"),
         Farm_type == "Broiler") %>%
  t.test(Observed ~ Sample_type, 
       data = .
       )


##Comparing Shannon index
print("SHANNON PULLET")
phycount_div %>%
  filter(Sample_type %in% c("Soil", "Litter"),
         Farm_type == "Pullet") %>%
  t.test(Shannon ~ Sample_type, 
       data = .
       )
print("SHANNON BREEDER")
phycount_div %>%
  filter(Sample_type %in% c("Soil", "Litter"),
         Farm_type == "Breeder") %>%
  t.test(Shannon ~ Sample_type, 
       data = .
       )
print("SHANNON BROILER")
phycount_div %>%
  filter(Sample_type %in% c("Soil", "Litter"),
         Farm_type == "Broiler") %>%
  t.test(Shannon ~ Sample_type, 
       data = .
       )

##Comparing Simpson index
print("SIMPSON PULLET")
phycount_div %>%
  filter(Sample_type %in% c("Soil", "Litter"),
         Farm_type == "Pullet") %>%
  t.test(Simpson ~ Sample_type, 
       data = .
       )
print("SIMPSON BREEDER")
phycount_div %>%
  filter(Sample_type %in% c("Soil", "Litter"),
         Farm_type == "Breeder") %>%
  t.test(Simpson ~ Sample_type, 
       data = .
       )
print("SIMPSON BROILER")
phycount_div %>%
  filter(Sample_type %in% c("Soil", "Litter"),
         Farm_type == "Broiler") %>%
  t.test(Observed ~ Sample_type, 
       data = .
       )


```

```{R writing alpha div ttest to file}

##Saving results (CAUTION!Run ONLY WHEN READY! Leave commented to avoid overwriting already generated files)
## Sink command, it print everything betweeN sinks()

set.seed(43)

##----------------SINK STARTS

# sink("AlphaDiversity_ttest.txt")
# 
#   ##Comparing OBserved Richness
#   print("RICHNESS PULLET")
#   phycount_div %>%
#     filter(Sample_type %in% c("Soil", "Litter"),
#            Farm_type == "Pullet") %>%
#     t.test(Observed ~ Sample_type, 
#          data = .
#          )
#   print("RICHNESS BREEDER")
#   phycount_div %>%
#     filter(Sample_type %in% c("Soil", "Litter"),
#            Farm_type == "Breeder") %>%
#     t.test(Observed ~ Sample_type, 
#          data = .
#          )
#   print("RICHNESS BROILER")
#   phycount_div %>%
#     filter(Sample_type %in% c("Soil", "Litter"),
#            Farm_type == "Broiler") %>%
#     t.test(Observed ~ Sample_type, 
#          data = .
#          )
#   
#   
#   ##Comparing Shannon index
#   print("SHANNON PULLET")
#   phycount_div %>%
#     filter(Sample_type %in% c("Soil", "Litter"),
#            Farm_type == "Pullet") %>%
#     t.test(Shannon ~ Sample_type, 
#          data = .
#          )
#   print("SHANNON BREEDER")
#   phycount_div %>%
#     filter(Sample_type %in% c("Soil", "Litter"),
#            Farm_type == "Breeder") %>%
#     t.test(Shannon ~ Sample_type, 
#          data = .
#          )
#   print("SHANNON BROILER")
#   phycount_div %>%
#     filter(Sample_type %in% c("Soil", "Litter"),
#            Farm_type == "Broiler") %>%
#     t.test(Shannon ~ Sample_type, 
#          data = .
#          )
#   
#   ##Comparing Simpson index
#   print("SIMPSON PULLET")
#   phycount_div %>%
#     filter(Sample_type %in% c("Soil", "Litter"),
#            Farm_type == "Pullet") %>%
#     t.test(Simpson ~ Sample_type, 
#          data = .
#          )
#   print("SIMPSON BREEDER")
#   phycount_div %>%
#     filter(Sample_type %in% c("Soil", "Litter"),
#            Farm_type == "Breeder") %>%
#     t.test(Simpson ~ Sample_type, 
#          data = .
#          )
#   print("SIMPSON BROILER")
#   phycount_div %>%
#     filter(Sample_type %in% c("Soil", "Litter"),
#            Farm_type == "Broiler") %>%
#     t.test(Observed ~ Sample_type, 
#          data = .
#          )
# 
# sink()

##----------------SINK ENDS

```


Drawing Richness and Diversity plots

```{r boxplot rich $ diversity}
#Creating vector for comparing samples
 complist <- list(c("Soil", "Litter"))

t.test()

#Creating vector with new names
 Farm_name <- c("Pullet", "Breeder", "Broiler", "Processing\nPlant")
names(Farm_name) <- c("Pullet", "Breeder", "Broiler", "Processing_plant")


#Drawing Observed Richness plot
bp_rc <- 
      phycount_div %>%
  # filter(Sample_type == "Litter" | Sample_type == "Soil") %>%
  filter(Farm_type != "Processing_plant" & Sample_type != "Fecal") %>%
  ggplot(aes(x = Sample_type,
             y = Observed, 
             fill = Sample_type
             )
         ) +
  geom_boxplot() +
  ggtitle("Observed Richness") +
  theme_bw() +
  theme(axis.text = element_text(size = 12, 
                                 color = "black", 
                                 hjust = 0.5
                                   ),
        axis.text.x = element_text(angle = 45,
                                   vjust = 1,
                                   hjust = 1),
        axis.title = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_line(color = "azure2"),
        plot.title = element_text(size = 14, face = "bold"),
        strip.background = element_rect(fill = "transparent",
                                        color = "transparent"),
        strip.text = element_text(size = 12,
                                  face = "bold"
                                  )
        ) +
  stat_compare_means(comparisons = complist, 
                     method = "t.test", 
                     label = "p.signif", 
                     vjust = 1
                     ) +
  scale_fill_manual(values = col_sample) +
  #facet_grid(Farm_type ~ Company,
   facet_grid(Farm_type ~ .,
             labeller = labeller(Farm_type = Farm_name),
             scales = "free",
            #space = "free_x"
             )



#Drawing Shannon diversity index plot
bp_sn <- 
    phycount_div %>%
  # filter(Sample_type == "Litter" | Sample_type == "Soil") %>%
  filter(Farm_type != "Processing_plant" & Sample_type != "Fecal") %>%
  ggplot(aes(x = Sample_type, 
             y = Shannon, 
             fill = Sample_type
             )
         ) +
  geom_boxplot() +
  ggtitle("Shannon Diversity Index") +
  theme_bw() +
  theme(axis.text = element_text(size = 12, 
                                 color = "black", 
                                 hjust = 0.5
                                   ),
        axis.text.x = element_text(angle = 45,
                                   vjust = 1,
                                   hjust = 1),
        axis.title = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_line(color = "azure2"),
        plot.title = element_text(size = 14, face = "bold"),
        strip.background = element_rect(fill = "transparent",
                                        color = "transparent"),
        strip.text = element_text(size = 12,
                                  face = "bold"
                                  )
        ) +
  stat_compare_means(comparisons = complist, 
                     method = "t.test", 
                     label = "p.signif", 
                     vjust = 1
                     ) +
  scale_fill_manual(values = col_sample) +
  #facet_grid(Farm_type ~ Company,
  facet_grid(Farm_type ~ .,
             labeller = labeller(Farm_type = Farm_name),
             scales = "free",
            space = "free_x"
             )


#Drawing Simpson diversity index plot
bp_sp <- 
  phycount_div %>%
  # filter(Sample_type == "Litter" | Sample_type == "Soil") %>%
  filter(Farm_type != "Processing_plant" & Sample_type != "Fecal") %>%
  ggplot(aes(x = Sample_type,
             y = Simpson, 
             fill = Sample_type
                    )
                ) +
  geom_boxplot() +
  ggtitle("Simpson Diversity Index") +
  theme_bw() +
  theme(axis.text = element_text(size = 12, 
                                 color = "black", 
                                 hjust = 0.5
                                   ),
        axis.text.x = element_text(angle = 45,
                                   vjust = 1,
                                   hjust = 1),
        axis.title = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_line(color = "azure2"),
        plot.title = element_text(size = 14, face = "bold"),
        strip.background = element_rect(fill = "transparent",
                                        color = "transparent"),
        strip.text = element_text(size = 12,
                                  face = "bold"
                                  )
        ) +
  stat_compare_means(comparisons = complist, 
                     method = "t.test", 
                     label = "p.signif", 
                     vjust = 1
                     ) +
  scale_fill_manual(values = col_sample) +
  #facet_grid(Farm_type ~ Company,
  facet_grid(Farm_type ~ .,
             labeller = labeller(Farm_type = Farm_name),
             scales = "free",
            space = "free_x"
             )


```  

```{r drawing alpha diversity box plots}
bp_rc
bp_sn
bp_sp
```

```{r saving bp plots}
#Saving plots (CAUTION!Run ONLY WHEN READY! Leave commented to avoid overwriting already generated plots)

# ggsave(filename = "BoxPlot_Richness_Litter-Soil.svg",
#        plot = bp_rc,
#        device = "svg",
#        units = "mm",
#        width = 150,
#        height = 159)
# 
# ggsave(filename = "BoxPlot_Shannon_Litter-Soil.svg",
#        plot = bp_sn,
#        device = "svg",
#        units = "mm",
#        width = 150,
#        height = 159)
# 
# ggsave(filename = "BoxPlot_Simpson_Litter-Soil.svg",
#        plot = bp_sp,
#        device = "svg",
#        units = "mm",
#        width = 150,
#        height = 159)

```

ADD INTERPRETATION____

Farm environment: Alpha diversity indexes show the significant differences between litter and soil.

      A. Observed richness: it was higher in soil than litter in both companies. But there was exception where in company B litter was slightly higher than soil, but not statistically significant. This indicates number different taxons/species are more in soil samples than litter samples.
      B. Sahnnon diversity index: It was higher in soil than litter. for both companies. This indicates..................
      C. Simpson diversity index: It was higher in soil than litter. for both companies. This indicates .........................

Processing plant: Alpha diversity was higher for all three indexes in post pick than post chill stage. This indicates reduction in species diversity following the chilling process. Note: it is obvious for diversity and microbial load to be reduced in post pick stage because of all the sanitation steps that take place during processing of chicken.   
    
Statistically significant differences were observed (check the images of plots)  
  
  
Conclusion: _______
  



##Exploring the microbial community




```{r metaphlan taxa}
table(tax_table(phycount_e)[,"Phylum"])
```


  
###Community bar plots

First step in visualizing the community structure.
In order to create the barplots, the data needs to be transformed from wide to long. This will also allow all the metadata to be incorporated into the table

```{r creating data frame}

#Melting table

tbl_mstr_pc <- melt(rownames_to_column(tbl_otu, var = "Taxon"),
                 id.vars = c("Taxon"),
                 variable.name = "Sample",
                 value.name = "Count") %>%
#Adding Metadata
            left_join(mt_smpl,
                      by = "Sample") %>%
            left_join(rownames_to_column(mt_tax, var = "Taxon"),
                      by = "Taxon")

```

Re-ordering factors so that the main colors of the swatch correspond to the most abundant phyla across:

```{r reordering}
#Summerizing counts by phyla

tbl_mstr_pc %>% 
  group_by(Phylum) %>% 
  summarise(Count = sum(Count)) %>% 
  arrange(Count) %>%
  select(Phylum) %>%
  ungroup()

#Reordering factors in Phylum column according to most abundant.
  # NOTE: Phyla will be reordered in ascending order (from least to most). This allows us to drop some levels (ex: Unclassified), without changing the color key for the other levels

tbl_mstr_pc$Phylum <- factor(tbl_mstr_pc$Phylum, 
                          levels = c("Deferribacteres","Rhodothermaeota","Candidatus_Adlerbacteria","Candidatus_Saccharibacteria","Lentisphaerae","Spirochaetes","Chlamydiae","Synergistetes","Verrucomicrobia","Candidatus_Thermoplasmatota","Gemmatimonadetes","Nitrospirae","Candidatus_Melainabacteria","Bacteria_unclassified","Fusobacteria","Acidobacteria","Planctomycetes","Chloroflexi","Deinococcus_Thermus","Cyanobacteria","Bacteroidetes","Proteobacteria","Actinobacteria","Firmicutes","Ascomycota","Thaumarchaeota","Euryarchaeota","UNCLASSIFIED"))

```

Creating summarised table for barplots
  -->NOTE: If table is not summarised, i.e. taxa summed *before* plotting, the result is a barplot filled with thin ligther lines. That happens as every single SGB is stacked on top of each other as a strip with white borders. When 300 are stacked the lines become apparent and degrade the graph.
        SOLUTION: Eithr sum before OR set color = "transparent"
  
```{r creating barplot table}

tbl_mstr_bp <- tbl_mstr_pc %>%
  group_by(Sample, Phylum) %>%
  summarise(Count = sum(Count)
            ) %>%
  left_join(mt_smpl, by = "Sample") %>%
  distinct() %>%
  droplevels() %>%
  ungroup

```

```{r drawing bar plots}

relbp_all <- 
  tbl_mstr_bp %>%
  filter(Sample_type!="Post_pick" & Sample_type!="Post_chill") %>%
  ggplot(aes(x = Sample,
           y = Count,
           fill = Phylum
           )
       ) +
  geom_bar(position="fill", stat= "identity") +
  scale_y_continuous(labels = percent) +
  scale_fill_manual(values = col_phy) +
  theme_bw() +
  labs(title = "Microbiome Composition") +
  ylab("Relative Abundance") +
  theme_bw() +
  theme(axis.text = element_text(size = 12,
                                 color = "black"),
        axis.text.x = element_text(hjust = 1,
                                 vjust = 0.5,
                                 angle = 90),
        axis.title = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_line(color = "azure2"),
        plot.title = element_text(size = 14, face = "bold"),
        strip.background = element_rect(fill = "transparent",
                                        color = "transparent"),
        strip.text = element_text(size = 12,
                                  face = "bold"
                                  )
        ) +
    facet_wrap( Sample_type ~ Farm_type, 
              scales = "free", 
              nrow = 4,
              ncol = 3
              )
 # facet_grid(Farm_type ~ Company, scales = "free", space = "free_x") 

  
  
  
# subset(tbl_mstr_bp,Phylum!="UNCLASSIFIED"),

relbp_cla <- 
  tbl_mstr_bp %>%
  filter(Phylum!="UNCLASSIFIED") %>%
  filter(Sample_type!="Post_pick" & Sample_type!="Post_chill") %>%
  ggplot(aes(x = Sample, 
             y = Count, 
             fill = Phylum 
             )
         ) + 
  geom_bar(position="fill", stat= "identity") +
  scale_y_continuous(labels = percent) +
  scale_fill_manual(values = col_phy) +
  theme_bw() + 
  labs(title = "Microbiome Composition - Classified") +
  ylab("Relative Abundance") +
  theme_bw() +
  theme(axis.text = element_text(size = 12, 
                                 color = "black"),
        axis.text.x = element_text(hjust = 1,
                                 vjust = 0.5,
                                 angle = 90), 
        axis.title = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_line(color = "azure2"),
        plot.title = element_text(size = 14, face = "bold"),
        strip.background = element_rect(fill = "transparent",
                                        color = "transparent"),
        strip.text = element_text(size = 12,
                                  face = "bold"
                                  )
        ) +
    facet_wrap( Sample_type ~ Farm_type, 
              scales = "free", 
              nrow = 4,
              ncol = 3
              )
#facet_grid(Farm_type ~ Company, scales = "free", space = "free_x")


relbp_bac <- 
  tbl_mstr_bp %>% 
  filter(Phylum!="UNCLASSIFIED" & Phylum!="Thaumarchaeota" & Phylum!="Euryarchaeota" & Phylum!="Ascomycota") %>%
  #filter(Company=="A") %>%
  filter(Sample_type!="Post_pick" & Sample_type!="Post_chill") %>%
  ggplot(aes(x = Sample, 
             y = Count, 
             fill = Phylum
             )
         ) + 
  geom_bar(position="fill", stat= "identity") +
  scale_y_continuous(labels = percent) +
  scale_fill_manual(values = col_phy) +
  theme_bw() + 
  labs(title = "Microbiome Composition - Bacterial") +
  ylab("Relative Abundance") +
  theme_bw() +
  theme(axis.text = element_text(size = 12, 
                                 color = "black"),
        axis.text.x = element_text(hjust = 1,
                                 vjust = 0.5,
                                 angle = 90), 
        axis.title = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_line(color = "azure2"),
        plot.title = element_text(size = 14, face = "bold"),
        strip.background = element_rect(fill = "transparent",
                                        color = "transparent"),
        strip.text = element_text(size = 12,
                                  face = "bold"
                                  )
        ) +
  facet_wrap( Sample_type ~ Farm_type, 
              scales = "free", 
              nrow = 4,
              ncol = 3
              )

```

```{r drawing relab plots}

relbp_all
relbp_cla
relbp_bac


```
Saving plots

```{r saving relative abundance plots}

#Saving plots (CAUTION!Run ONLY WHEN READY! Leave commented to avoid overwriting already generated plots)

# ggsave(filename = "Barplot_RelAb_All.svg",
#        plot = relbp_all,
#        device = "svg",
#        units = "mm",
#        width = 500,
#         height = 300)
#  
#  ggsave(filename = "Barplot_RelAb_Classified.svg",
#         plot = relbp_cla,
#         device = "svg",
#         units = "mm",
#         width = 500,
#         height = 300)
#  
#  ggsave(filename = "Barplot_RelAb_Bacterial.svg",
#         plot = relbp_bac,
#         device = "svg",
#         units = "mm",
#        width = 500,
#        height = 300)

```

These plots show the relative abundance of the taxa per sample replicate. For increased robustness, the replicates will averaged at taxon (SGB) level

```{r creating averaged table}
# Averaging taxons by Sample_group

tbl_mstr_av <- tbl_mstr_pc %>%
  group_by(Taxon, Sample_group) %>%
  summarise(Count = mean(Count)) %>%
  left_join(select(mt_smpl, !c(Sample, TotalReads)), by = "Sample_group") %>%
  left_join(rownames_to_column(mt_tax, var = "Taxon"), by = "Taxon") %>%
  distinct() %>%
  droplevels() %>%
  ungroup()



##Summarising by Phylum count

tbl_mstr_av_bp <- tbl_mstr_av %>%
  group_by(Phylum, Sample_group) %>%
  summarise(Count = sum(Count)) %>%
  left_join(select(mt_smpl, !c(Sample, TotalReads)), by = "Sample_group") %>%
  distinct() %>%
  droplevels() %>%
  ungroup()

#Reordering Phylum factors (left join brought in factors with disordered levels)

tbl_mstr_av$Phylum <- factor(tbl_mstr_av$Phylum, 
                          levels = c("Deferribacteres","Rhodothermaeota","Candidatus_Adlerbacteria","Candidatus_Saccharibacteria","Lentisphaerae","Spirochaetes","Chlamydiae","Synergistetes","Verrucomicrobia","Candidatus_Thermoplasmatota","Gemmatimonadetes","Nitrospirae","Candidatus_Melainabacteria","Bacteria_unclassified","Fusobacteria","Acidobacteria","Planctomycetes","Chloroflexi","Deinococcus_Thermus","Cyanobacteria","Bacteroidetes","Proteobacteria","Actinobacteria","Firmicutes","Ascomycota","Thaumarchaeota","Euryarchaeota","UNCLASSIFIED"))




tbl_mstr_av_bp$Phylum <- factor(tbl_mstr_av_bp$Phylum, 
                           levels = c("Deferribacteres","Rhodothermaeota","Candidatus_Adlerbacteria","Candidatus_Saccharibacteria","Lentisphaerae","Spirochaetes","Chlamydiae","Synergistetes","Verrucomicrobia","Candidatus_Thermoplasmatota","Gemmatimonadetes","Nitrospirae","Candidatus_Melainabacteria","Bacteria_unclassified","Fusobacteria","Acidobacteria","Planctomycetes","Chloroflexi","Deinococcus_Thermus","Cyanobacteria","Bacteroidetes","Proteobacteria","Actinobacteria","Firmicutes","Ascomycota","Thaumarchaeota","Euryarchaeota","UNCLASSIFIED"))

```





```{r drawing AVERAGED bar plots}

relbp_av_all <- 
  tbl_mstr_av_bp %>%
  subset(Sample_type != "Fecal" & Farm_type != "Processing_plant") %>%
  ggplot(aes(x = Sample_group, 
             y = Count, 
             fill = Phylum
             )
         ) + 
  geom_bar(position="fill", stat= "identity") +
  scale_y_continuous(labels = percent) +
  scale_fill_manual(values = col_phy) +
  theme_bw() +
  labs(title = "Microbiome Composition") +
  ylab("Relative Abundance") +
  theme_bw() +
  theme(axis.text = element_text(size = 12,
                                 color = "black"),
        axis.text.x = element_text(hjust = 1,
                                 vjust = 0.5,
                                 angle = 90),
        axis.title = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_line(color = "azure2"),
        plot.title = element_text(size = 14, face = "bold"),
        strip.background = element_rect(fill = "transparent",
                                        color = "transparent"),
        strip.text = element_text(size = 12,
                                  face = "bold"
                                  )
        ) +
  facet_grid(Sample_type ~ Farm_type, scales = "free", space = "free")
  #facet_wrap(Company ~ Sample_type ~ Farm_type,
             #scales = "free",
              #nrow = 4,
              #ncol = 3
            #  )


relbp_av_cla <- 
 tbl_mstr_av_bp %>%
  subset(Phylum!="UNCLASSIFIED" & Sample_type != "Fecal" & Farm_type != "Processing_plant") %>%
  ggplot(aes(x = Sample_group, 
             y = Count, 
             fill = Phylum
             )
         ) + 
  geom_bar(position="fill", stat= "identity") +
  scale_y_continuous(labels = percent) +
  scale_fill_manual(values = col_phy) +
  theme_bw() + 
  labs(title = "Microbiome Composition - Classified") +
  ylab("Relative Abundance") +
  theme_bw() +
  theme(axis.text = element_text(size = 12, 
                                 color = "black"),
        axis.text.x = element_text(hjust = 1,
                                 vjust = 0.5,
                                 angle = 90), 
        axis.title = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_line(color = "azure2"),
        plot.title = element_text(size = 14, face = "bold"),
        strip.background = element_rect(fill = "transparent",
                                        color = "transparent"),
        strip.text = element_text(size = 12,
                                  face = "bold"
                                  )
        ) +
  facet_grid(Sample_type ~ Farm_type, scales = "free", space = "free")
    #facet_wrap(Company ~ Sample_type ~ Farm_type,
     #         scales = "free",
      #        nrow = 4,
       #       ncol = 3
          #    )

relbp_av_bac <- 
  tbl_mstr_av_bp %>%
  subset(Phylum!="UNCLASSIFIED" & Phylum!="Thaumarchaeota" & Phylum!= "Euryarchaeota" & Phylum!= "Ascomycota" & Sample_type != "Fecal" & Farm_type != "Processing_plant") %>%
  ggplot(aes(x = Sample_group, 
             y = Count, 
             fill = Phylum
             )
         ) + 
  geom_bar(position="fill", stat= "identity") +
  scale_y_continuous(labels = percent) +
  scale_fill_manual(values = col_phy) +
  theme_bw() + 
  labs(title = "Microbiome Composition - Bacterial") +
  ylab("Relative Abundance") +
  theme_bw() +
  theme(axis.text = element_text(size = 12, 
                                 color = "black"),
        axis.text.x = element_text(hjust = 1,
                                 vjust = 0.5,
                                 angle = 90), 
        axis.title = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_line(color = "azure2"),
        plot.title = element_text(size = 14, face = "bold"),
        strip.background = element_rect(fill = "transparent",
                                        color = "transparent"),
        strip.text = element_text(size = 12,
                                  face = "bold"
                                  )
        ) +
    facet_grid(Sample_type ~ Farm_type, scale = "free", space = "free")
 
    # facet_wrap(Company ~ Sample_type ~ Farm_type,
    #           scales = "free",
    #           nrow = 4,
    #           ncol = 3
    #           )



### seperating according to sample type because with facetwrap and facet grid i couldnt get the structure the way i wanted.

####Litter bac

relbp_av_bac_Litter <- 
tbl_mstr_av_bp %>%
  subset(Phylum!="UNCLASSIFIED" & Phylum!="Thaumarchaeota" & Phylum!= "Euryarchaeota" & Phylum!= "Ascomycota" & Sample_type == "Litter"  & Farm_type != "Processing_plant") %>%
  ggplot(aes(x = Sample_group, 
             y = Count, 
             fill = Phylum
             )
         ) + 
  geom_bar(position="fill", stat= "identity") +
  scale_y_continuous(labels = percent) +
  scale_fill_manual(values = col_phy) +
  theme_bw() + 
  labs(title = "Microbiome Composition - Bacterial_Litter") +
  ylab("Relative Abundance") +
  theme_bw() +
  theme(axis.text = element_text(size = 12, 
                                 color = "black"),
        axis.text.x = element_text(hjust = 1,
                                 vjust = 0.5,
                                 angle = 90), 
        axis.title = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_line(color = "azure2"),
        plot.title = element_text(size = 14, face = "bold"),
        strip.background = element_rect(fill = "transparent",
                                        color = "transparent"),
        strip.text = element_text(size = 12,
                                  face = "bold"
                                  )
        ) +
  facet_grid(Sample_type ~ Farm_type, scale = "free_x", space = "free_x")



####Soil bac

relbp_av_bac_Soil <- 
tbl_mstr_av_bp %>%
  subset(Phylum!="UNCLASSIFIED" & Phylum!="Thaumarchaeota" & Phylum!= "Euryarchaeota" & Phylum!= "Ascomycota" & Sample_type == "Soil"  & Farm_type != "Processing_plant") %>%
  ggplot(aes(x = Sample_group, 
             y = Count, 
             fill = Phylum
             )
         ) + 
  geom_bar(position="fill", stat= "identity") +
  scale_y_continuous(labels = percent) +
  scale_fill_manual(values = col_phy) +
  theme_bw() + 
  labs(title = "Microbiome Composition - Bacterial_Soil") +
  ylab("Relative Abundance") +
  theme_bw() +
  theme(axis.text = element_text(size = 12, 
                                 color = "black"),
        axis.text.x = element_text(hjust = 1,
                                 vjust = 0.5,
                                 angle = 90), 
        axis.title = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_line(color = "azure2"),
        plot.title = element_text(size = 14, face = "bold"),
        strip.background = element_rect(fill = "transparent",
                                        color = "transparent"),
        strip.text = element_text(size = 12,
                                  face = "bold"
                                  )
        ) +
  facet_grid(Sample_type ~ Farm_type, scale = "free_x", space = "free_x")
```


```{r drawing AVERAGED bar plots_PP}
#Processing Plant
# Note: Subsetting using ! command is different than inclusion command. For inclusion in eed to use %in% followed by all things I want to include inside bracket.
relbp_av_bac_PP <- 
  tbl_mstr_av_bp %>%
  subset(Phylum!="UNCLASSIFIED" & Phylum!="Thaumarchaeota" & Phylum!= "Euryarchaeota" & Phylum!= "Ascomycota" & Sample_group %in% c("C4F8PP", "C4F8PC", "C8F40L", "C8F40S")) %>%
  ggplot(aes(x = Sample_group, 
             y = Count, 
             fill = Phylum
             )
         ) + 
  geom_bar(position="fill", stat= "identity") +
  scale_y_continuous(labels = percent) +
  scale_fill_manual(values = col_phy) +
  theme_bw() + 
  labs(title = "Microbiome Composition - Bacterial") +
  ylab("Relative Abundance") +
  theme_bw() +
  theme(axis.text = element_text(size = 12, 
                                 color = "black"),
        axis.text.x = element_text(hjust = 1,
                                 vjust = 0.5,
                                 angle = 90), 
        axis.title = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_line(color = "azure2"),
        plot.title = element_text(size = 14, face = "bold"),
        strip.background = element_rect(fill = "transparent",
                                        color = "transparent"),
        strip.text = element_text(size = 12,
                                  face = "bold"
                                  )
        ) +
    facet_grid(Sample_type ~ Farm_type, scale = "free", space = "free")
       #facet_wrap(Company ~ Sample_type ~ Farm_type,
        #     scales = "free",
         #     nrow = 4,
          #    ncol = 3
           #   )



  

```

```{r drawing averaged relab plots}

relbp_av_all
relbp_av_cla
relbp_av_bac
# relbp_av_bac_combined
relbp_av_bac_Litter
relbp_av_bac_Soil
relbp_av_bac_PP

```
Saving plots

```{r saving averaged relative abundance plots}

# #Saving plots (CAUTION!Run ONLY WHEN READY! Leave commented to avoid overwriting already generated plots)

# ggsave(filename = "Barplot_RelAbAV_All.svg",
#        plot = relbp_av_all,
#        device = "svg",
#        units = "mm",
#        width = 240,
#        height = 350)
# 
# ggsave(filename = "Barplot_RelAbAv_Classified.svg",
#        plot = relbp_av_cla,
#        device = "svg",
#        units = "mm",
#        width = 240,
#        height = 350)
# 
# ggsave(filename = "Barplot_RelAbAv_Bacterial.svg",
#        plot = relbp_av_bac,
#        device = "svg",
#        units = "mm",
#        width = 240,
#        height = 350)
# 
# 
# 
# ggsave(filename = "Barplot_RelAbAv_Bacterial_Litter.svg",
#        plot = relbp_av_bac_Litter,
#        device = "svg",
#        units = "mm",
#        width = 500,
#        height = 350)
# 
# 
# ggsave(filename = "Barplot_RelAbAv_Bacterial_Soil.svg",
#        plot = relbp_av_bac_Soil,
#        device = "svg",
#        units = "mm",
#        width = 500,
#        height = 350)
# 
# 
# 
# ggsave(filename = "Barplot_RelAbAv_Bacterial_PP.svg",
#        plot = relbp_av_bac_PP,
#        device = "svg",
#        units = "mm",
#        width = 500,
#        height = 350)
# 

```



  The bar plots above align the initial assumptions about these two environments:
    --> Soil has a markedly different microbiome profile with much higher phyla diversity than that found in the bee microbiome. 
    Let's count the number of Taxa present in each Source
    
 
 
## Remember our OTU counts are converted to be between 0-1. 



##Getting table for percent microbe in plots

```{r percent count from plots}
# Generate percent composition table ensuring each Sample_group sums to 100% and removing 0% values
percent_composition_table <- tbl_mstr_av_bp %>%
  filter(Phylum != "UNCLASSIFIED", 
         Phylum != "Thaumarchaeota", 
         Phylum != "Euryarchaeota", 
         Phylum != "Ascomycota",
         Sample_type != "Fecal"  & Farm_type != "Processing_plant") %>% 
  group_by(Sample_group, Farm_type, Sample_type, Company, Phylum) %>%
  summarise(Total_Count = sum(Count), .groups = "drop") %>%
  group_by(Sample_group, Farm_type, Sample_type, Company) %>%
  mutate(Percent_Composition = (Total_Count / sum(Total_Count)) * 100) %>%
  filter(Percent_Composition > 0) %>%  # Remove Phyla with 0% abundance
  arrange(Sample_group, Farm_type, Sample_type, Company, desc(Percent_Composition)) %>% 
  ungroup()


##Save everything in one CSV file
##write.csv(percent_composition_table, "percent_composition_all_sample_groups.csv", row.names = FALSE)


```









### Exploring Beta-Diversity

  If alpha-diversity is the diversity *within* a community, beta-diversity is the diversity across communities. In beta-diversity we are concerned with comparing and contrasting different profiles.

  Communities can be statistically compared by producing distance matrices. On compositional data we will use robust Aitchison distances, which relies on the centered log ratio transform. Once these distances are calculated, profiles can be statistically compared using PERMANOVA and visualized using ordination plots, such as PCoA or PCA.
  
  In order to perform these analysis the abundances will be again exported into phyloseq but this time as relative abundances. The phyloseq object will be used as input into two different packages:
      --> Vegan: where PERMANOVA and PERMAdisp will be calculates
      --> MicrobiomeMarker: where Ordinations and normalization will be performed
      
```{r creating relative abundance phyloseq object}

OTUcount = otu_table(tbl_otu, 
                taxa_are_rows = TRUE)
      #Needs to be parsed as matrix.
TAX = tax_table(as.matrix(mt_tax)
                )
      #Needs to be parsed as matrix.
sampledata = sample_data(mt_smpl)

#Creating phyloseq objects
phycount = phyloseq(OTUcount, TAX, sampledata)
      #Tree doesn't need to be processed. Direct import

#Excluding Unclassified, Archaea and Dropping incomplete sample groups: Larvae and Pollen
phycount <- phycount %>%
  subset_taxa(!Phylum == "UNCLASSIFIED" & !Kingdom == "Archaea" & !Kingdom == "Eukaryota") %>%
  subset_samples(!Sample_type == "Fecal" & !Farm_type == "Processing_plant")


phycount_e %>%
  

```

```{r printing phyloseq summary}
phycount

print("")
print("Unique Phyla in phycount Object")
get_taxa_unique(phycount, "Phylum")
print("Sample Sources in phycount object")
levels(sample_data(phycount)$Sample_type)
levels(sample_data(phycount)$Farm_type)

```
##Performing PERMANOVAS

--- from: https://cscu.cornell.edu/workshop/introduction-to-permanova/
PERMANOVA, (permutational multivariate ANOVA), is a non-parametric alternative to MANOVA, or multivariate ANOVA test. It is appropriate with multiple sets of variables that do not meet the assumptions of MANOVA, namely multivariate normality. It is often used with data that is highly skewed, zero-inflated, ordinal, or qualitative, such as ecological community data, microbiome data, or genetic data.

It operates on a distance matrix constructed from any dissimilarity measure, and tests the null hypothesis that there are no differences in the relative magnitude (or presence/absence) of a set of variables among objects from different treatments or groups. P-values are calculated via permutation tests.  PERMANOVA tests often accompany ordination plots from PCoA or NMDS.

---

  For calculating our distance matrices, Aithcison distances will be employed since the are equivalent to euclidean-distances of CLR transformed data. As discussed int he beginning of this document, the robust Aitchison distance will be employed since it is applicable to non-negative data inclusing zeroes (which is the case of microbiome data)
(https://rdrr.io/cran/vegan/man/vegdist.html).

  PERMANOVA sensitivity is dependent on group dispersion. A significant result in a PERMANOVA can be attributed to either groups differing in their location (on an ordination), their dispersion or both. In order to understand what is driving the differences between groups the dispersions will be checked using PERMDISP: a multivariate Levene's test that checks the homogeneity of dispersion. "A non-significant result from PERMDISP would indicate that groups do not differ in dispersion and that therefore the differences are entirely due to differences in location.  A significant result from PERMDISP would indicate that the groups differ in dispersion." But in the later case they could *also* differ in location, thus visual inspection would be necessary.
  >NOTE: Permutational techniques do not assume homogeneity of variances. PERMANOVA is safe to be employed even if dispersion (variances) are not homogenous. The test is necessary to better interpret the results

For more in depth explanations on PERMANOVA: https://uw.pressbooks.pub/appliedmultivariatestatistics/chapter/permanova/

For more in depth explanations on PERMDISP: https://uw.pressbooks.pub/appliedmultivariatestatistics/chapter/permdisp/

PERMANOVAS and PERMDISP are calculated using the adonis and betadisper functions in the package Vegan. To use Vegan, phyloseq objects need to be turned into Vegan objects.
  
  Next functions convert physeq format into Vegan. Source: https://jacobrprice.github.io/2017/08/26/phyloseq-to-vegan-and-back.html
  >NOTE: The fuctions keep refering to OTU since all of these packages were designed to work with 16S data. SGBs are functionally similar to OTUs for our purpuses.

```{r creating pssd2veg function}

pssd2veg <- function(physeq) {
  sd <- sample_data(physeq)
  return(as(sd,"data.frame"))
}

```

```{r creating psotu2veg }

psotu2veg <- function(physeq) {
  OTU <- otu_table(physeq)
  if (taxa_are_rows(OTU)) {
    OTU <- t(OTU)
  }
  return(as(OTU, "matrix"))
}

```

Importing Phyloseq data into teh package Vegan

```{r data import }

vegan_count <- pssd2veg(phycount_e)
vegan_otu <- psotu2veg(phycount_e)

```

Calculating distance matrix

```{r distance matrix}
##Calculating distance matrix using Robust Aitchinson (Euclidean distance of the CLR transform of non-zero values)
veg_count_raitch <- vegdist(vegan_otu, "robust.aitchison")


```

###RUNNING PERMDISP

Calculating Multivariate homogeneity of groups dispersions

```{r running PERMDISP}
###Calculating distances and Comparing dispersions

set.seed(43) #Set seed allows the random permutations to be reproducible

    ##--> Group = Source
print("---Estimating Dispersion for $Sample_type---", quote = FALSE, justify = "centre")
betadisper(veg_count_raitch, 
           vegan_count$Sample_type,
           type = "centroid")
print("", quote = FALSE)
print("", quote = FALSE)
print("---Comparing Source dispersions by permutation test---", quote = FALSE, justify = "centre")
permutest(betadisper(veg_count_raitch, 
                     vegan_count$Sample_type), 
          pairwise = TRUE)
print("------xxx------", quote = FALSE, justify = "centre")
print("", quote = FALSE)
print("", quote = FALSE)


###Calculating distances and Comparing dispersions
    ##--> Group = Farm_type
print("---Estimating Dispersion for $Farm_type---", quote = FALSE, justify = "centre")
betadisper(veg_count_raitch, 
           vegan_count$Farm_type,
           type = "centroid")
print("", quote = FALSE)
print("", quote = FALSE)
print("---Comparing Farm_type disprsions by permutation test---", quote = FALSE, justify = "centre")
permutest(betadisper(veg_count_raitch, 
                     vegan_count$Farm_type),
          pairwise = TRUE)
print("------xxx------", quote = FALSE, justify = "centre")
print("", quote = FALSE)
print("", quote = FALSE)


###Calculating distances and Comparing dispersions
    ##--> Group = Company
print("---Estimating Dispersion for $Company---", quote = FALSE, justify = "centre")
betadisper(veg_count_raitch, 
           vegan_count$Company,
           type = "centroid")
print("", quote = FALSE)
print("", quote = FALSE)
print("---Comparing Company betadisper by permutation test---", quote = FALSE, justify = "centre")
permutest(betadisper(veg_count_raitch, 
                     vegan_count$Company), 
          pairwise = TRUE)
print("------xxx------", quote = FALSE, justify = "centre")
print("", quote = FALSE)
print("", quote = FALSE)


###Calculating distances and Comparing dispersions
    ##--> Group = Sample_group
print("---Estimating Dispersion for $Sample_group---", quote = FALSE, justify = "centre")
betadisper(veg_count_raitch, 
           vegan_count$Sample_group,
           type = "centroid")
print("", quote = FALSE)
print("", quote = FALSE)
print("---Comparing Sample_group betadisper by permutation test---", quote = FALSE, justify = "centre")
permutest(betadisper(veg_count_raitch, 
                     vegan_count$Sample_group),
          pairwise = FALSE)
print("------xxx------", quote = FALSE, justify = "centre")

```




```{r plotting PERMDISP dispersions}
# Plotting dispersions
plot(betadisper(veg_count_raitch, vegan_count$Sample_type))
plot(betadisper(veg_count_raitch, vegan_count$Farm_type))
plot(betadisper(veg_count_raitch, vegan_count$Company))
plot(betadisper(veg_count_raitch, vegan_count$Sample_group))


```

PERMDISP analysis shows that:

 
###RUNNING PERMANOVAs

```{r running distance PERMANOVA}

set.seed(43)

print("Comparing profiles by PERMANOVA",quote = FALSE, justify = "centre")
print("Grouping by Sample_group (Sample_type*Farm_type*Company",quote = FALSE, justify = "centre")

adonis2(veg_count_raitch ~ Sample_type*Farm_type*Company, data = vegan_count, permutations = 999)
print("",quote = FALSE, justify = "centre")
print("Grouping by Sample_type",quote = FALSE, justify = "centre")

adonis2(veg_count_raitch ~ Sample_type, data = vegan_count, permutations = 999)
print("",quote = FALSE, justify = "centre")
print("Grouping by Farm_type",quote = FALSE, justify = "centre")

adonis2(veg_count_raitch ~ Farm_type, data = vegan_count, permutations = 999)
print("",quote = FALSE, justify = "centre")
print("Grouping by Company",quote = FALSE, justify = "centre")

adonis2(veg_count_raitch ~ Company, data = vegan_count, permutations = 999)

```

```{r specific adonis pariwise}
## Install pairwiseAdonis package if not already installed
if (!requireNamespace("devtools", quietly = TRUE)) {
 install.packages("devtools")
}
devtools::install_github("pmartinezarbizu/pairwiseAdonis/pairwiseAdonis")

# Load the package
library(pairwiseAdonis)

# Perform pairwise PERMANOVA for Sample_type
pairwise.adonis2(veg_count_raitch ~ Sample_type, data = vegan_count, permutations = 999)

# Perform pairwise PERMANOVA for Farm_type
pairwise.adonis2(veg_count_raitch ~ Farm_type, data = vegan_count, permutations = 999)

```





### ORDINATION WITH PCOA

  For the ordination we will use PCoA with robust Aitchison distances of CLR transforms, which (for reasons I honestly do not fully understand statistically) is the same thing as performing a PCA on a CLR transformed data. PCA + rCLR as the go-to ordination for compositional data, has been championed and commonly applied to COmpositional Data Analysis (CODA) (https://www.frontiersin.org/journals/microbiology/articles/10.3389/fmicb.2017.02224/full).
  However, I'd be remiss if I did not bring up that there has been objections to this practice, arguing that a rCLR transform is not enough to allow the data to be handled by PCA and might introduce biases. The most powerfull way to deal with microbial compositional data is the Isometric Log-ratio transform (ILR). However, I could not find good implementation and pipelines using it, and thus seems to require much heavier and deeper statistical knowledge to implement (which I don't have). Nonetheless, below are some papers that did use ILR for more information:
    > https://journals.asm.org/doi/10.1128/msystems.00162-16
    > https://peerj.com/articles/2969/
    > https://elifesciences.org/articles/21887

Drawing PCA plots

```{r drawing PCoA + Aithcison plots}

##As per the instructions in the man page, data needs to be UNORMALIZED!Attempting to normalize relative abundances with CLR will create negative values for abundances 0< x<1 and break on zeros.
##As per the instructions in the man page, data needs to be passed as is 'trans="identity"'. Dist calc robust.aitchison will calculate robust CLR and compute distance

#Sample type
pcoa_sp_clr <-
  phycount_e %>%
 tax_transform(rank = "Species", 
               trans = "identity") %>%  
 dist_calc(dist = "robust.aitchison") %>%
 ord_calc(method = "PCoA") %>% 
 ord_plot(axes = c(1, 2),
          plot_taxa = 1:3,
          colour = "black", 
          fill = "Farm_type",
          shape = "Sample_type",
          alpha = 0.8,
          size = 5
          ) + 
  stat_ellipse(aes(colour = Sample_type), linewidth = 0.3) +
  scale_shape_girafe_filled() +
  ggtitle("PCoA by Species-level Genomic Bins") +
   guides(fill = guide_legend(override.aes=list(shape = 21)),
          color = FALSE) +
  scale_fill_manual(values = col_farm) +
  # scale_color_manual(values = col_sample) +
  # scale_alpha_discrete(range = c(0.35, 1)) +
  theme_linedraw() +
  theme(panel.grid = element_blank(),
        panel.background = element_rect(fill = "#fdfdfd"),
        axis.text = element_text(size = 14,
                                 color = "black"),
        axis.title = element_text(size = 16,
                                  color = "black")) +
  geom_text(x = 2.5, 
            y = -3.4, 
            hjust =0,
            vjust = 1,
            label="Sample Type:\nPERMANOVA=0.001\nPERMDISP=0.001",
            size = 3.5,
            fontface = "plain") #+
  #facet_grid(. ~ Company, scale = "free")


pcoa_sp_clr_S <-
  phycount_e %>%
 tax_transform(rank = "Species", 
               trans = "identity") %>%  
 dist_calc(dist = "robust.aitchison") %>%
 ord_calc(method = "PCoA") %>% 
 ord_plot(axes = c(1, 2),
          plot_taxa = 1:3,
          colour = "black", 
          fill = "Sample_type",
          shape = "Farm_type",
          alpha = 0.8,
          size = 5
          ) + 
  stat_ellipse(aes(colour = Sample_type), linewidth = 0.3) +
  scale_shape_girafe_filled() +
  ggtitle("PCoA by Species-level Genomic Bins") +
   guides(fill = guide_legend(override.aes=list(shape = 21)),
          color = FALSE) +
  scale_fill_manual(values = col_sample) +
  # scale_color_manual(values = col_sample) +
  # scale_alpha_discrete(range = c(0.35, 1)) +
  theme_linedraw() +
  theme(panel.grid = element_blank(),
        panel.background = element_rect(fill = "#fdfdfd"),
        axis.text = element_text(size = 14,
                                 color = "black"),
        axis.title = element_text(size = 16,
                                  color = "black")) +
  geom_text(x = 2.5, 
            y = -3.4, 
            hjust =0,
            vjust = 1,
            label="Sample Type:\nPERMANOVA=0.001\nPERMDISP=0.001",
            size = 3.5,
            fontface = "plain") #+
  #facet_grid(. ~ Company, scale = "free")


#Farm type
pcoa_sp_clr_farm <-
  phycount_e %>%
 tax_transform(rank = "Species", 
               trans = "identity") %>%  
 dist_calc(dist = "robust.aitchison") %>%
 ord_calc(method = "PCoA") %>% 
 ord_plot(axes = c(1, 2),
          plot_taxa = 1:3,
          colour = "black", 
          fill = "Farm_type",
          shape = "Sample_type",
          alpha = 0.8,
          size = 5
          ) + 
  stat_ellipse(aes(colour = Farm_type), linewidth = 0.3) +
  scale_shape_girafe_filled() +
  ggtitle("PCoA by Species-level Genomic Bins") +
   guides(fill = guide_legend(override.aes=list(shape = 21)),
          color = FALSE) +
  scale_fill_manual(values = col_farm) +
  scale_color_manual(values = col_farm) +
  # scale_alpha_discrete(range = c(0.35, 1)) +
  theme_linedraw() +
  theme(panel.grid = element_blank(),
        panel.background = element_rect(fill = "#fdfdfd"),
        axis.text = element_text(size = 14,
                                 color = "black"),
        axis.title = element_text(size = 16,
                                  color = "black")) +
  geom_text(x = 2.5, 
            y = -3.4, 
            hjust =0,
            vjust = 1,
            label="Farm Type:\nPERMANOVA=0.001\nPERMDISP=0.191\nCompany:\nPERMANOVA=0.009\nPERMDISP=0.081",
            size = 3.5,
            fontface = "plain") #+
  #facet_grid(. ~ Company, scale = "free")


pcoa_sp_clr_S_farm <-
  phycount_e %>%
 tax_transform(rank = "Species", 
               trans = "identity") %>%  
 dist_calc(dist = "robust.aitchison") %>%
 ord_calc(method = "PCoA") %>% 
 ord_plot(axes = c(1, 2),
          plot_taxa = 1:3,
          colour = "black", 
          fill = "Sample_type",
          shape = "Farm_type",
          alpha = 0.8,
          size = 5
          ) + 
  stat_ellipse(aes(colour = Farm_type), linewidth = 0.3) +
  scale_shape_girafe_filled() +
  ggtitle("PCoA by Species-level Genomic Bins") +
   guides(fill = guide_legend(override.aes=list(shape = 21)),
          color = FALSE) +
  # scale_fill_manual(values = col_farm) +
  # scale_color_manual(values = col_farm) +
  # scale_alpha_discrete(range = c(0.35, 1)) +
  theme_linedraw() +
  theme(panel.grid = element_blank(),
        panel.background = element_rect(fill = "#fdfdfd"),
        axis.text = element_text(size = 14,
                                 color = "black"),
        axis.title = element_text(size = 16,
                                  color = "black")) +
  geom_text(x = 2.5, 
            y = -3.4, 
            hjust =0,
            vjust = 1,
            label="Farm Type:\nPERMANOVA=0.001\nPERMDISP=0.191\nCompany:\nPERMANOVA=0.009\nPERMDISP=0.081",
            size = 3.5,
            fontface = "plain") #+
  #facet_grid(. ~ Company, scale = "free")




```



```{r drawing PCoA plots}

pcoa_sp_clr
pcoa_sp_clr_S

pcoa_sp_clr_farm
pcoa_sp_clr_S_farm

```

```{r saving PCoA}
#Saving plots (CAUTION!Run ONLY WHEN READY! Leave commented to avoid overwriting already generated plots)

 # ggsave(filename = "PCoA_Microbiome_SourceEllip_SampleColor_SAMPLETYPE.svg",
 #        plot = pcoa_sp_clr,
 #        device = "svg",        
 #        units = "mm",
 #        width = 300,
 #        height = 200)
 # 
 # ggsave(filename = "PCoA_Microbiome_SourceEllip_FarmColor_SAMPLETYPE.svg",
 #        plot = pcoa_sp_clr_S,
 #        device = "svg",        
 #        units = "mm",
 #        width = 300,
 #        height = 200)
 # 
 #  ggsave(filename = "PCoA_Microbiome_SourceEllip_SampleColor_FARMTYPE.svg",
 #        plot = pcoa_sp_clr_farm,
 #        device = "svg",        
 #        units = "mm",
 #        width = 300,
 #        height = 200)
 # 
 # ggsave(filename = "PCoA_Microbiome_SourceEllip_FarmColor_FARMTYPE.svg",
 #        plot = pcoa_sp_clr_S_farm,
 #        device = "svg",        
 #        units = "mm",
 #        width = 300,
 #        height = 200)


```

The ordination shows that .....


PERMDISP analysis shows that.......................




#################################################################################